% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/annotations.R
\name{import_omnipath_annotations}
\alias{import_omnipath_annotations}
\alias{import_Omnipath_annotations}
\alias{import_OmniPath_annotations}
\title{Imports annotations from OmniPath}
\usage{
import_omnipath_annotations(
  proteins = NULL,
  resources = NULL,
  wide = FALSE,
  organism = "human",
  genesymbol_resource = NULL,
  ...
)
}
\arguments{
\item{proteins}{Vector containing the genes or proteins for whom
annotations will be retrieved (UniProt IDs or HGNC Gene Symbols or
miRBase IDs). It is also possible to donwload annotations for protein
complexes. To do so, write 'COMPLEX:' right before the genesymbols of
the genes integrating the complex. Check the vignette for examples.}

\item{resources}{Load the annotations only from these databases.
See \code{\link{get_annotation_resources}} for possible values.}

\item{wide}{Convert the annotation table to wide format, which
corresponds more or less to the original resource. If the data comes
from more than one resource a list of wide tables will be returned.
See examples at \code{\link{pivot_annotations}}.}

\item{organism}{Character or integer: Name or NCBI Taxonomy ID of one or
organisms. The web service currently provides intercell annotations
only for human. For other organisms, the data will be translated
by orthologous gene pairs from human.}

\item{genesymbol_resource}{Character: either "uniprot" or "ensembl". The
former leaves intact the gene symbols returned by the web service,
originally set from UniProt. The latter updates the gene symbols from
Ensembl, which uses a slightly different gene symbol standard. In this
case a few records will be duplicated, where Ensembl provides ambiguous
translation.}

\item{...}{Additional arguments.}
}
\value{
A data frame containing different gene and complex annotations.
}
\description{
Imports protein annotations about function, localization, expression,
structure and other properties of proteins from OmniPath
\url{https://omnipathdb.org/annotations}.
Note: there might be also a few miRNAs annotated; a vast majority of
protein complex annotations are inferred from the annotations of the
members: if all members carry the same annotation the complex inherits.
}
\details{
Downloading the full annotations
dataset is disabled by default because the size of this data is
around 1GB. We recommend to retrieve the annotations for a set of proteins
or only from a few resources, depending on your interest. You can always
download the full database from
\url{https://archive.omnipathdb.org/omnipath_webservice_annotations__recent.tsv}
using any standard R or \code{readr} method.
}
\examples{
annotations <- import_omnipath_annotations(
    proteins = c('TP53', 'LMNA'),
    resources = c('HPA_subcellular')
)

}
\seealso{
\itemize{
    \item{\code{\link{get_annotation_databases}}}
    \item{\code{\link{pivot_annotations}}}
}
}
