<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="OmnipathR">
<title>Building prior knowledge network (PKN) for COSMOS • OmnipathR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Building prior knowledge network (PKN) for COSMOS">
<meta property="og:description" content="OmnipathR">
<meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">OmnipathR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.13.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    <a class="dropdown-item" href="../articles/cosmos.html">Building prior knowledge network (PKN) for COSMOS</a>
    <a class="dropdown-item" href="../articles/db_manager.html">The database manager in OmnipathR</a>
    <a class="dropdown-item" href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    <a class="dropdown-item" href="../articles/extra_attrs.html">Resource specific interaction attributes</a>
    <a class="dropdown-item" href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    <a class="dropdown-item" href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    <a class="dropdown-item" href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
    <a class="dropdown-item" href="../articles/paths.html">Pathway construction</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/OmnipathR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Building prior knowledge network (PKN) for COSMOS</h1>
                        <h4 data-toc-skip class="author">Denes
Turei</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:turei.denes@gmail.com" class="email">turei.denes@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/HEAD/vignettes/cosmos.Rmd" class="external-link"><code>vignettes/cosmos.Rmd</code></a></small>
      <div class="d-none name"><code>cosmos.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The prior knowledge network (PKN) used by COSMOS is a network
      of heterogenous causal interactions: it contains protein-protein,
      reactant-enzyme and enzyme-product interactions. It is a
      combination of multiple resources. Here we present the functions
      that load each component, the options for customization, and the
      functions to build the complete PKN. </p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The COSMOS PKN is a combination of the following datasets:</p>
<ul>
<li>Genome-scale metabolic model (GEM) from Chalmers Sysbio (Wang et
al., 2021.)</li>
<li>Network of chemical-protein interactions from STITCH (<a href="http://stitch.embl.de/" class="external-link uri">http://stitch.embl.de/</a>)</li>
<li>Protein-protein interactions from Omnipath (Türei et al., 2021)</li>
</ul>
<p>Building the PKN is possible in by calling the
<code><a href="../reference/cosmos_pkn.html">cosmos_pkn()</a></code> function that we present in the last section.
First let’s take a closer look at each resource.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.omnipathdb.org/" class="external-link">OmnipathR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="chalmers-sysbio-gem">Chalmers Sysbio GEM<a class="anchor" aria-label="anchor" href="#chalmers-sysbio-gem"></a>
</h2>
<p>The Chalmers Sysbio group provides genome scale models of metabolism
(GEMs) for various organisms: human, mouse, rat, fish, fly and worm.
These models are available as Matlab files, which contain reaction data,
stoichiometry matrix and identifier translation data.</p>
<p>The raw contents of the Matlab file can be loaded by
<code><a href="../reference/chalmers_gem_raw.html">chalmers_gem_raw()</a></code>. This results in a convoluted structure
of nested lists and arrays:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chalmers_gem_raw.html">chalmers_gem_raw</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Another function, <code><a href="../reference/chalmers_gem.html">chalmers_gem()</a></code> processes the above
structure into a data frame of reactions, also keeping the stoichiometry
matrix and including the identifier translation data:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_processed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chalmers_gem.html">chalmers_gem</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: Matrix</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ch_processed</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "organism"       "reactions"      "metabolites"    "reaction_ids"  </span></span>
<span><span class="co">## [5] "metabolite_ids" "S"</span></span></code></pre>
<p>The identifier translation tables are available by separate
functions, <code><a href="../reference/chalmers_gem_metabolites.html">chalmers_gem_metabolites()</a></code> and
<code><a href="../reference/chalmers_gem_reactions.html">chalmers_gem_reactions()</a></code> for metabolite and reaction (and
enzyme) identifiers, respectively. These return simple data frames.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_met</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chalmers_gem_metabolites.html">chalmers_gem_metabolites</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ch_met</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 8,456 × 14</span></span></span>
<span><span class="co">##    mets      metsNoComp metBiGGID metKEGGID metHMDBID   metChEBIID  metPubChemID</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> MAM00001c MAM00001   carveol   C00964    <span style="color: #BB0000;">NA</span>          CHEBI:15389           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> MAM00001e MAM00001   carveol   C00964    <span style="color: #BB0000;">NA</span>          CHEBI:15389           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> MAM00002c MAM00002   appnn     C09880    HMDB0006525 CHEBI:36740         <span style="text-decoration: underline;">6</span>654</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> MAM00002e MAM00002   appnn     C09880    HMDB0006525 CHEBI:36740         <span style="text-decoration: underline;">6</span>654</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> MAM00003c MAM00003   <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          CHEBI:78990           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MAM00003l MAM00003   <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          CHEBI:78990           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MAM00003r MAM00003   <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          CHEBI:78990           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MAM00003e MAM00003   <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          CHEBI:78990           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MAM00004c MAM00004   <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>                    <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> MAM00004m MAM00004   <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>        <span style="color: #BB0000;">NA</span>          <span style="color: #BB0000;">NA</span>                    <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 8,446 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 7 more variables: metLipidMapsID &lt;chr&gt;, metEHMNID &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   metHepatoNET1ID &lt;chr&gt;, metRecon3DID &lt;chr&gt;, metMetaNetXID &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   metHMR2ID &lt;chr&gt;, metRetired &lt;chr&gt;</span></span></span></code></pre>
<p>The metabolite identifier translation available here is also
integrated into the package’s translation service, available by the
<code>translate_ids</code> and other functions.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/translate_ids.html">translate_ids</a></span><span class="op">(</span><span class="st">'MAM00001'</span>, <span class="st">'metabolicatlas'</span>, <span class="st">'recon3d'</span>, chalmers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "carveol"</span></span></code></pre>
<p>Finally, the <code><a href="../reference/chalmers_gem_network.html">chalmers_gem_network()</a></code> function uses all
above data to compile a network binary chemical-protein interactions. By
default Metabolic Atlas identifiers are used for the metabolites and
Ensembl Gene IDs for the enzymes. These can be tranlated to the desired
identifiers using the <code>metabolite_ids</code> and
<code>protein_ids</code> arguments. Translation to multiple identifers
is possible. The <code>ri</code> or <code>record_id</code> column in
case of the Chalmers GEM represent the reaction ID, a unique identifier
of the original reaction. One reaction yields many binary interactions
as it consists of a number of gene products, reactants and products. The
column <code>ci</code> means “complex ID”, it is a unique identifier of
groups of enzymes required together to carry out the reaction. The
column <code>reverse</code> indicates if the row is derived from the
reversed version of a reversible reaction. The column
<code>transporter</code> signals reactions where the same metabolite
occures both on reactant and product side, these are assumed to be
transport reactions. In the Chalmers GEM the reactions are also assigned
to compartments, these are encoded by single letter codes in the
<code>comp</code> column. In the original data the compartment codes are
postfixes of the metabolite IDs, here we move them into a separate
column, leaving the Metabolic Atlas IDs clean and usable.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chalmers_gem_network.html">chalmers_gem_network</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ch</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 135,769 × 16</span></span></span>
<span><span class="co">##       ri    ci source   target          reverse met_to_gene comp  transporter</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>     1     1 MAM01796 ENSG00000147576 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>     1     1 MAM01796 ENSG00000147576 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>     1     2 MAM01796 ENSG00000172955 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>     1     2 MAM01796 ENSG00000172955 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>     1     3 MAM01796 ENSG00000180011 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>     1     3 MAM01796 ENSG00000180011 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>     1     4 MAM01796 ENSG00000187758 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>     1     5 MAM01796 ENSG00000196344 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>     1     5 MAM01796 ENSG00000196344 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>     1     6 MAM01796 ENSG00000196616 FALSE   TRUE        c     <span style="color: #BB0000;">NA</span>         </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 135,759 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 8 more variables: uniprot_source &lt;chr&gt;, genesymbol_source &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   uniprot_target &lt;chr&gt;, genesymbol_target &lt;chr&gt;, hmdb_source &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   kegg_source &lt;chr&gt;, hmdb_target &lt;chr&gt;, kegg_target &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="stitch-enzyme-metabolite-interactions">STITCH enzyme-metabolite interactions<a class="anchor" aria-label="anchor" href="#stitch-enzyme-metabolite-interactions"></a>
</h2>
<p>STITCH is a large compendium of binary interactions between proteins
and chemicals. Some of these are derived from metabolic reactions.
Various attributes such as mode of action, effect sign and scores are
assigned to each link. The datasets are available by organism, stored in
“actions” and “links” tables, available by the
<code><a href="../reference/stitch_actions.html">stitch_actions()</a></code> and <code><a href="../reference/stitch_links.html">stitch_links()</a></code> functions,
respectively. STITCH supports a broad variety of organisms, please refer
to their website at (<a href="https://stitch.embl.de/" class="external-link uri">https://stitch.embl.de/</a>).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sta</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stitch_actions.html">stitch_actions</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sta</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 21,773,491 × 6</span></span></span>
<span><span class="co">##    item_id_a       item_id_b       mode       action a_is_acting score</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ENSP00000170630 10461           expression <span style="color: #BB0000;">NA</span>     FALSE         150</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> 10461           ENSP00000170630 expression <span style="color: #BB0000;">NA</span>     TRUE          150</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ENSP00000353915 23627457        binding    <span style="color: #BB0000;">NA</span>     FALSE         191</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 23627457        ENSP00000353915 binding    <span style="color: #BB0000;">NA</span>     FALSE         191</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ENSP00000256906 44408029        binding    <span style="color: #BB0000;">NA</span>     FALSE         521</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 44408029        ENSP00000256906 binding    <span style="color: #BB0000;">NA</span>     FALSE         521</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ENSP00000267377 23590374        pred_bind  <span style="color: #BB0000;">NA</span>     FALSE         170</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 23590374        ENSP00000267377 pred_bind  <span style="color: #BB0000;">NA</span>     FALSE         170</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ENSP00000267377 23590374        binding    <span style="color: #BB0000;">NA</span>     FALSE         159</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> 23590374        ENSP00000267377 binding    <span style="color: #BB0000;">NA</span>     FALSE         159</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 21,773,481 more rows</span></span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stitch_links.html">stitch_links</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">stl</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 15,473,939 × 7</span></span></span>
<span><span class="co">##    chemical protein   experimental prediction database textmining combined_score</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> 91758680 ENSP0000…            0          0        0        278            279</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> 91758680 ENSP0000…            0          0        0        154            154</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> 91758408 ENSP0000…            0          0        0        225            225</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 91758408 ENSP0000…            0          0        0        178            178</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> 91758408 ENSP0000…            0          0        0        225            225</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 91758408 ENSP0000…            0          0        0        151            151</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> 91758408 ENSP0000…            0          0        0        162            162</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 91758408 ENSP0000…            0          0        0        194            194</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> 91758408 ENSP0000…            0          0        0        169            169</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> 91758408 ENSP0000…            0          0        0        163            163</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 15,473,929 more rows</span></span></span></code></pre>
<p><code><a href="../reference/stitch_gem.html">stitch_gem()</a></code> combines the actions and links data frames,
filters by confidence scores, removes the prefixes from identifiers and
translates them to the desired ID types. STITCH prefixes Ensembl Protein
IDs with the NCBI Taxonomy ID, while PubChem CIDs with CID plus a
lowercase letter “s” or “m”, meaning stereo specific or merged
stereoisomers, respectively. These prefixes are removed by default by
the <code><a href="../reference/stitch_remove_prefixes.html">stitch_remove_prefixes()</a></code> function. Effect signs (1 =
activation, -1 = inhibition) and combined scores are aslo included in
the data frame. Similarly to Chalmers GEM, translation of chemical and
protein identifiers is available. The <code>record_id</code> column
uniquely identifies the original records. Multiple rows with the same
<code>record_id</code> are due to one-to-many identifier
translation.</p>
<pre class="ri"><code><span><span class="va">st</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stitch_gem.html">stitch_gem</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">st</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="omnipath-signaling-network">OmniPath signaling network<a class="anchor" aria-label="anchor" href="#omnipath-signaling-network"></a>
</h2>
<p>All parameters supported by the OmniPath web service
(<code><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions()</a></code>) can be passed to
<code><a href="../reference/omnipath_for_cosmos.html">omnipath_for_cosmos()</a></code>, enabling precise control over the
resources, interaction types and other options when preparing the
signaling network from OmniPath. By default the “omnipath” dataset is
included which contains high confidence, literature curated, causal
protein-protein interactions. For human, mouse and rat, orthology
translated data is retrieved from the web service, while for other
organisms translation by orthologous gene pairs is carried out client
side.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">op</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/omnipath_for_cosmos.html">omnipath_for_cosmos</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">op</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 141,695 × 8</span></span></span>
<span><span class="co">##    source target  sign record_id uniprot_source uniprot_target genesymbol_source</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> P0DP23 P48995    -<span style="color: #BB0000;">1</span>         1 P0DP23         P48995         CALM1            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> P0DP25 P48995    -<span style="color: #BB0000;">1</span>         2 P0DP25         P48995         CALM3            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> P0DP25 P48995    -<span style="color: #BB0000;">1</span>         2 P0DP25         P48995         CALM1            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> P0DP24 P48995    -<span style="color: #BB0000;">1</span>         3 P0DP24         P48995         CALM2            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> P0DP24 P48995    -<span style="color: #BB0000;">1</span>         3 P0DP24         P48995         CALM1            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Q03135 P48995     1         4 Q03135         P48995         CAV1             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> P14416 P48995     1         5 P14416         P48995         DRD2             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Q99750 P48995    -<span style="color: #BB0000;">1</span>         6 Q99750         P48995         MDFI             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Q14571 P48995     1         7 Q14571         P48995         ITPR2            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> P29966 P48995    -<span style="color: #BB0000;">1</span>         8 P29966         P48995         MARCKS           </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 141,685 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: genesymbol_target &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="complete-build">Complete build<a class="anchor" aria-label="anchor" href="#complete-build"></a>
</h2>
<p>Building the complete COSMOS PKN is done by
<code><a href="../reference/cosmos_pkn.html">cosmos_pkn()</a></code>. All the resources above can be customized by
arguments passed to this function. With all downloads and processing the
build might take 30-40 minutes. Data is cached at various levels of
processing, shortening processing times. With all data downloaded and
HMDB ID translation data preprocessed, the build takes 3-4 minutes; the
complete PKN is also saved in the cache, if this is available, loading
it takes only a few seconds.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cosmos_pkn.html">cosmos_pkn</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pkn</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 323,716 × 20</span></span></span>
<span><span class="co">##    source  target  sign record_id resource entity_type_source entity_type_target</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> MAM017… ENSG0…    <span style="color: #BB0000;">NA</span>         1 Chalmer… metabolite         protein           </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 323,706 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 13 more variables: score_stitch &lt;dbl&gt;, ci_chalmers &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   comp_chalmers &lt;fct&gt;, reverse_chalmers &lt;lgl&gt;, transporter_chalmers &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   uniprot_source &lt;chr&gt;, genesymbol_source &lt;chr&gt;, uniprot_target &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   genesymbol_target &lt;chr&gt;, hmdb_source &lt;chr&gt;, kegg_source &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   hmdb_target &lt;chr&gt;, kegg_target &lt;chr&gt;</span></span></span></code></pre>
<p>The <code>record_id</code> column identifies the original records
within each resource. If one <code>record_id</code> yields multiple
records in the final data frame, it is the result of one-to-many ID
translation or other processing steps. Before use, it is recommended to
select one pair of ID type columns (by combining the preferred ones) and
perform <code>distinct</code> by the identifier columns and sign. After
the common columns, resource specific columns are labeled with the
resource name; after these columns, molecule type and side specific
identifer columns are named after the ID type and the side of the
interaction (“source” vs. “target”).</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Arch Linux</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/libblas.so.3.12.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/liblapack.so.3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Madrid</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] Matrix_1.6-5     OmnipathR_3.13.1 BiocStyle_2.30.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] xfun_0.42           bslib_0.6.1         lattice_0.22-5     </span></span>
<span><span class="co">##  [4] tzdb_0.4.0          vctrs_0.6.5         tools_4.3.3        </span></span>
<span><span class="co">##  [7] generics_0.1.3      curl_5.2.1          tibble_3.2.1       </span></span>
<span><span class="co">## [10] fansi_1.0.6         R.oo_1.25.0         pkgconfig_2.0.3    </span></span>
<span><span class="co">## [13] checkmate_2.3.1     desc_1.4.3          readxl_1.4.3       </span></span>
<span><span class="co">## [16] lifecycle_1.0.4     compiler_4.3.3      stringr_1.5.1      </span></span>
<span><span class="co">## [19] textshaping_0.3.7   progress_1.2.3      htmltools_0.5.7    </span></span>
<span><span class="co">## [22] sass_0.4.9          yaml_2.3.8          later_1.3.2        </span></span>
<span><span class="co">## [25] pillar_1.9.0        pkgdown_2.0.7       crayon_1.5.2       </span></span>
<span><span class="co">## [28] jquerylib_0.1.4     tidyr_1.3.1         R.utils_2.12.3     </span></span>
<span><span class="co">## [31] cachem_1.0.8        tidyselect_1.2.1    rvest_1.0.4        </span></span>
<span><span class="co">## [34] zip_2.3.0           digest_0.6.35       stringi_1.8.3      </span></span>
<span><span class="co">## [37] dplyr_1.1.4         purrr_1.0.2         bookdown_0.38      </span></span>
<span><span class="co">## [40] grid_4.3.3          fastmap_1.1.1       cli_3.6.2          </span></span>
<span><span class="co">## [43] logger_0.3.0        magrittr_2.0.3      XML_3.99-0.16.1    </span></span>
<span><span class="co">## [46] utf8_1.2.4          readr_2.1.5         withr_3.0.0        </span></span>
<span><span class="co">## [49] prettyunits_1.2.0   backports_1.4.1     rappdirs_0.3.3     </span></span>
<span><span class="co">## [52] bit64_4.0.5         lubridate_1.9.3     timechange_0.3.0   </span></span>
<span><span class="co">## [55] rmarkdown_2.26      httr_1.4.7          igraph_2.0.3       </span></span>
<span><span class="co">## [58] bit_4.0.5           R.matlab_3.7.0      cellranger_1.1.0   </span></span>
<span><span class="co">## [61] R.methodsS3_1.8.2   ragg_1.2.7          hms_1.1.3          </span></span>
<span><span class="co">## [64] memoise_2.0.1       evaluate_0.23       knitr_1.45         </span></span>
<span><span class="co">## [67] rlang_1.1.3         Rcpp_1.0.12         glue_1.7.0         </span></span>
<span><span class="co">## [70] selectr_0.4-2       BiocManager_1.30.22 xml2_1.3.6         </span></span>
<span><span class="co">## [73] vroom_1.6.5         jsonlite_1.8.8      R6_2.5.1           </span></span>
<span><span class="co">## [76] systemfonts_1.0.6   fs_1.6.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor, Diego Mananes, Aurelien Dugourd.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
