<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="OmnipathR">
<title>Resource specific interaction attributes • OmnipathR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Resource specific interaction attributes">
<meta property="og:description" content="OmnipathR">
<meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">OmnipathR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.9.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    <a class="dropdown-item" href="../articles/db_manager.html">The database manager in OmnipathR</a>
    <a class="dropdown-item" href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    <a class="dropdown-item" href="../articles/extra_attrs.html">Resource specific interaction attributes</a>
    <a class="dropdown-item" href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    <a class="dropdown-item" href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    <a class="dropdown-item" href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
    <a class="dropdown-item" href="../articles/paths.html">Pathway construction</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/OmnipathR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Resource specific interaction attributes</h1>
                        <h4 data-toc-skip class="author">Denes
Turei</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:turei.denes@gmail.com" class="email">turei.denes@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/HEAD/vignettes/extra_attrs.Rmd" class="external-link"><code>vignettes/extra_attrs.Rmd</code></a></small>
      <div class="d-none name"><code>extra_attrs.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>OmniPath provides a broad variety of protein annotations, but
      for interactions, until recently, only a standard set of essential
      attributes (direction, effect, etc) and a handful of others
      (e.g. DoRothEA confidence level) were available. The newly
      introduced <code>extra_attrs</code> column consists of JSON
      encoded custom, resource specific attributes from network
      databases. We also revised the processing of these resources to
      ensure that we include as many useful attributes as possible. In
      the OmnipathR package we added a few new functions to support the
      processing of the JSON encoded column: to scan it for keys and
      values, and to extract specific variables of interest into new
      columns. We give a brief overview of these here.</p>
    </div>
    
<div class="section level2">
<h2 id="loading-a-network">Loading a network<a class="anchor" aria-label="anchor" href="#loading-a-network"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.omnipathdb.org/" class="external-link">OmnipathR</a></span><span class="op">)</span></span></code></pre></div>
<p>First we retrieve the complete directed PPI network. Importantly, the
extra attributes are only included if the
<code>fields = "extra_attrs"</code> argument is provided.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">i</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_post_translational_interactions.html">import_post_translational_interactions</a></span><span class="op">(</span>fields <span class="op">=</span> <span class="st">'extra_attrs'</span><span class="op">)</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">source_genesymbol</span>, <span class="va">target_genesymbol</span>, <span class="va">extra_attrs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 134,282 × 3</span></span></span>
<span><span class="co">##    source_genesymbol target_genesymbol extra_attrs     </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> CALM1             TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> CALM3             TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> CALM2             TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> CAV1              TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> DRD2              TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MDFI              TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ITPR2             TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MARCKS            TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> TRPC1             GRM1              <span style="color: #949494;">&lt;named list [0]&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> GRM1              TRPC1             <span style="color: #949494;">&lt;named list [1]&gt;</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 134,272 more rows</span></span></span></code></pre>
<p>Above we see, the <code>extra_attrs</code> column is a list type
column. Each list is a nested list itself, containing the extra
attributes from all resources, as it was extracted from the JSON.</p>
</div>
<div class="section level2">
<h2 id="which-extra-attributes-are-available">Which extra attributes are available?<a class="anchor" aria-label="anchor" href="#which-extra-attributes-are-available"></a>
</h2>
<p>Which attributes present in the network depends only on the
interactions: if none of the interactions is from the <code>SPIKE</code>
database, obviously the <code>SPIKE_mechanism</code> won’t be present.
The names of the extra attributes consist of the name of the resource
and the name of the attribute, separated by an underscore. The resource
name never contains underscore, while some attribute names do. To list
the extra attributes available in a particular data frame use the
<code>extra_attrs</code> function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/extra_attrs.html">extra_attrs</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "TRIP_method"                "SIGNOR_mechanism"          </span></span>
<span><span class="co">##  [3] "PhosphoSite_noref_evidence" "PhosphoPoint_category"     </span></span>
<span><span class="co">##  [5] "PhosphoSite_evidence"       "HPRD-phos_mechanism"       </span></span>
<span><span class="co">##  [7] "Li2012_mechanism"           "Li2012_route"              </span></span>
<span><span class="co">##  [9] "SPIKE_effect"               "SPIKE_mechanism"           </span></span>
<span><span class="co">## [11] "SPIKE_LC_effect"            "SPIKE_LC_mechanism"        </span></span>
<span><span class="co">## [13] "CA1_effect"                 "CA1_type"                  </span></span>
<span><span class="co">## [15] "Macrophage_type"            "Macrophage_location"       </span></span>
<span><span class="co">## [17] "ACSN_effect"                "Cellinker_type"            </span></span>
<span><span class="co">## [19] "CellChatDB_category"        "talklr_putative"           </span></span>
<span><span class="co">## [21] "CellPhoneDB_type"           "Ramilowski2015_source"     </span></span>
<span><span class="co">## [23] "HPMR_partner_role"          "ARN_effect"                </span></span>
<span><span class="co">## [25] "ARN_is_direct"              "ARN_is_directed"           </span></span>
<span><span class="co">## [27] "NRF2ome_effect"             "NRF2ome_is_direct"         </span></span>
<span><span class="co">## [29] "NRF2ome_is_directed"</span></span></code></pre>
<p>The labels listed here are the top level keys in the lists in the
<code>extra_attrs</code> column. Note, the coverage of these variables
varies a lot, typically in agreement with the size of the resource.</p>
</div>
<div class="section level2">
<h2 id="inspecting-one-attribute">Inspecting one attribute<a class="anchor" aria-label="anchor" href="#inspecting-one-attribute"></a>
</h2>
<p>The values of each extra attribute, in theory, can be arbitrarily
complex nested lists, but in reality, these are most often simple
numeric, logical or character values or vectors. To see the unique
values of one attribute use the <code>extra_attr_values</code> function.
Let’s see the values of the <code>SIGNOR_mechanism</code> attribute:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/extra_attr_values.html">extra_attr_values</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">SIGNOR_mechanism</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "phosphorylation"                    "binding"                           </span></span>
<span><span class="co">##  [3] "dephosphorylation"                  "Phosphorylation"                   </span></span>
<span><span class="co">##  [5] "ubiquitination"                     "N/A"                               </span></span>
<span><span class="co">##  [7] "Physical Interaction"               "Proteolytic Processing"            </span></span>
<span><span class="co">##  [9] "cleavage"                           "Ubiquitination"                    </span></span>
<span><span class="co">## [11] "Deubiqitination"                    "deubiquitination"                  </span></span>
<span><span class="co">## [13] "relocalization"                     "Dephosphorylation"                 </span></span>
<span><span class="co">## [15] "Other"                              "guanine nucleotide exchange factor"</span></span>
<span><span class="co">## [17] "Transcription Regulation"           "gtpase-activating protein"         </span></span>
<span><span class="co">## [19] "Indirect"                           ""                                  </span></span>
<span><span class="co">## [21] "Sumoylation"                        "sumoylation"                       </span></span>
<span><span class="co">## [23] "palmitoylation"                     "Acetylation"                       </span></span>
<span><span class="co">## [25] "acetylation"                        "polyubiquitination"                </span></span>
<span><span class="co">## [27] "Demethylation"                      "demethylation"                     </span></span>
<span><span class="co">## [29] "mRNA stability"                     "methylation"                       </span></span>
<span><span class="co">## [31] "Methylation"                        "trimethylation"                    </span></span>
<span><span class="co">## [33] "hydroxylation"                      "monoubiquitination"                </span></span>
<span><span class="co">## [35] "Deacetylation"                      "deacetylation"                     </span></span>
<span><span class="co">## [37] "Translational Regulation"           "Protein Degradation"               </span></span>
<span><span class="co">## [39] "Glycosylation"                      "s-nitrosylation"                   </span></span>
<span><span class="co">## [41] "phosphomotif_binding"               "chemical activation"               </span></span>
<span><span class="co">## [43] "Proteolytic Cleavage"               "tyrosination"                      </span></span>
<span><span class="co">## [45] "post transcriptional regulation"    "post translational modification"   </span></span>
<span><span class="co">## [47] "translation regulation"             "carboxylation"                     </span></span>
<span><span class="co">## [49] "neddylation"                        "Carboxylation"                     </span></span>
<span><span class="co">## [51] "desumoylation"                      "glycosylation"                     </span></span>
<span><span class="co">## [53] "ADP-ribosylation"                   "stabilization"                     </span></span>
<span><span class="co">## [55] "catalytic activity"                 "deglycosylation"                   </span></span>
<span><span class="co">## [57] "destabilization"                    "chemical inhibition"               </span></span>
<span><span class="co">## [59] "isomerization"                      "Neddylation"                       </span></span>
<span><span class="co">## [61] "lipidation"                         "chemical modification"             </span></span>
<span><span class="co">## [63] "oxidation"                          "Alkylation"</span></span></code></pre>
<p>The values are provided as they are in the original resource,
including potential typos and inconsistencies, e.g. see above the
capitalized vs. lowercase forms of each value.</p>
</div>
<div class="section level2">
<h2 id="converting-extra-attributes-to-columns">Converting extra attributes to columns<a class="anchor" aria-label="anchor" href="#converting-extra-attributes-to-columns"></a>
</h2>
<p>To make use of the attributes, it is convenient to extract the
interesting ones into separate columns of the data frame. With the
<code>extra_attrs_to_cols</code> function multiple attributes can be
converted in a single call. Custom column names can be passed by
argument names. As an example, let’s extract two attributes:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">i0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extra_attrs_to_cols.html">extra_attrs_to_cols</a></span><span class="op">(</span></span>
<span>    <span class="va">i</span>,</span>
<span>    si_mechanism <span class="op">=</span> <span class="va">SIGNOR_mechanism</span>,</span>
<span>    ma_mechanism <span class="op">=</span> <span class="va">Macrophage_type</span>,</span>
<span>    keep_empty <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>    <span class="va">i0</span>,</span>
<span>    <span class="va">source_genesymbol</span>,</span>
<span>    <span class="va">target_genesymbol</span>,</span>
<span>    <span class="va">si_mechanism</span>,</span>
<span>    <span class="va">ma_mechanism</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 61,406 × 4</span></span></span>
<span><span class="co">##    source_genesymbol target_genesymbol si_mechanism ma_mechanism</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> PRKG1             TRPC3             <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> PRKG1             TRPC7             <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> OS9               TRPV4             <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> PTPN1             TRPV6             <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> RACK1             TRPM6             <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> PRKACA            MCOLN1            <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MAPK14            MAPKAPK2          <span style="color: #949494;">&lt;list [1]&gt;</span>   <span style="color: #949494;">&lt;list [1]&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MAPKAPK2          HNRNPA0           <span style="color: #949494;">&lt;list [2]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MAPKAPK2          PARN              <span style="color: #949494;">&lt;list [2]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> JAK2              EPOR              <span style="color: #949494;">&lt;list [2]&gt;</span>   <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 61,396 more rows</span></span></span></code></pre>
<p>Above we disabled the <code>keep_empty</code> option, otherwise the
new columns would have <code>NULL</code> values for most of the records,
simply because out of the 80k interactions in the data frame only a few
thousands are from either SIGNOR or Macrophage. The new columns are list
type, individual values are character vectors. Let’s look into one
value:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">i0</span>, <span class="va">si_mechanism</span><span class="op">)</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [[1]]</span></span>
<span><span class="co">## [1] "phosphorylation"</span></span></code></pre>
<p>Here we have two values, but only because the inconsistent names in
the resource.</p>
<p>Depending on downstream methods, atomic columns might be preferable
instead of lists. In this case one interaction record might yield
multiple rows in the resulted data frame, depending on the number of
attributes it has. To have atomic columns, use the <code>flatten</code>
option:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">i1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extra_attrs_to_cols.html">extra_attrs_to_cols</a></span><span class="op">(</span></span>
<span>    <span class="va">i</span>,</span>
<span>    si_mechanism <span class="op">=</span> <span class="va">SIGNOR_mechanism</span>,</span>
<span>    ma_mechanism <span class="op">=</span> <span class="va">Macrophage_type</span>,</span>
<span>    keep_empty <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    flatten <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>    <span class="va">i1</span>,</span>
<span>    <span class="va">source_genesymbol</span>,</span>
<span>    <span class="va">target_genesymbol</span>,</span>
<span>    <span class="va">si_mechanism</span>,</span>
<span>    <span class="va">ma_mechanism</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 63,434 × 4</span></span></span>
<span><span class="co">##    source_genesymbol target_genesymbol si_mechanism ma_mechanism</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> PRKG1             TRPC3             <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> PRKG1             TRPC7             <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> OS9               TRPV4             <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> PTPN1             TRPV6             <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> RACK1             TRPM6             <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> PRKACA            MCOLN1            <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MAPK14            MAPKAPK2          <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;chr [1]&gt;</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MAPKAPK2          HNRNPA0           <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MAPKAPK2          HNRNPA0           <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> MAPKAPK2          PARN              <span style="color: #949494;">&lt;chr [1]&gt;</span>    <span style="color: #949494;">&lt;NULL&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 63,424 more rows</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="filtering-records-based-on-extra-attributes">Filtering records based on extra attributes<a class="anchor" aria-label="anchor" href="#filtering-records-based-on-extra-attributes"></a>
</h2>
<p>Another useful application of extra attributes is filtering the
records of the interactions data frame. The
<code>with_extra_attrs</code> function filters to records which have
certain extra attributes. For example, to have only interactions with
<code>SIGNOR_mechanism</code> given:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="../reference/with_extra_attrs.html">with_extra_attrs</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">SIGNOR_mechanism</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 61111</span></span></code></pre>
<p>This results around 11 thousands rows. Filtering for multiple
attributes the records which have at least one of them will be selected.
Adding some more attributes results more interactions:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="../reference/with_extra_attrs.html">with_extra_attrs</a></span><span class="op">(</span><span class="va">i</span>, <span class="va">SIGNOR_mechanism</span>, <span class="va">CA1_effect</span>, <span class="va">Li2012_mechanism</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 62017</span></span></code></pre>
<p>It is possible to filter the records not only by the names but the
values of the extra attributes. Let’s select the interactions which are
phosphorylation according to SIGNOR:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">phos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'phosphorylation'</span>, <span class="st">'Phosphorylation'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">si_phos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_extra_attrs.html">filter_extra_attrs</a></span><span class="op">(</span><span class="va">i</span>, SIGNOR_mechanism <span class="op">=</span> <span class="va">phos</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">si_phos</span>, <span class="va">source_genesymbol</span>, <span class="va">target_genesymbol</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 4,353 × 2</span></span></span>
<span><span class="co">##    source_genesymbol                                target_genesymbol</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> PRKG1                                            TRPC3            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> PRKG1                                            TRPC7            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> PRKACA                                           MCOLN1           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> MAPK14                                           MAPKAPK2         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> MAPKAPK2                                         HNRNPA0          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MAPKAPK2                                         PARN             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> JAK2                                             EPOR             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MAPK14                                           ZFP36            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MAPKAPK2                                         ZFP36            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> PRKAA1_PRKAA2_PRKAB1_PRKAB2_PRKAG1_PRKAG2_PRKAG3 CRTC2            </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4,343 more rows</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="example-finding-ubiquitination-interactions">Example: finding ubiquitination interactions<a class="anchor" aria-label="anchor" href="#example-finding-ubiquitination-interactions"></a>
</h2>
<p>First let’s search for the word “ubiquitination” in the attributes.
Below is a slow but simple solution:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keys</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extra_attrs.html">extra_attrs</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span><span class="va">keys_ubi</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/keep.html" class="external-link">keep</a></span><span class="op">(</span></span>
<span>    <span class="va">keys</span>,</span>
<span>    <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="../reference/extra_attr_values.html">extra_attr_values</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">!</span><span class="op">!</span><span class="va">k</span><span class="op">)</span>, <span class="st">'biqu'</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">)</span></span>
<span><span class="va">keys_ubi</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "SIGNOR_mechanism"    "HPRD-phos_mechanism" "SPIKE_mechanism"    </span></span>
<span><span class="co">## [4] "SPIKE_LC_mechanism"  "CA1_type"            "Macrophage_type"</span></span></code></pre>
<p>We found five attributes that have at least one value which matches
“biqu”. Next take a look at their values:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ubi</span> <span class="op">&lt;-</span> <span class="fu">rlang</span><span class="fu">::</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html" class="external-link">set_names</a></span><span class="op">(</span></span>
<span>    <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>        <span class="va">keys_ubi</span>,</span>
<span>        <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span><span class="op">{</span></span>
<span>            <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_subset.html" class="external-link">str_subset</a></span><span class="op">(</span><span class="fu"><a href="../reference/extra_attr_values.html">extra_attr_values</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">!</span><span class="op">!</span><span class="va">k</span><span class="op">)</span>, <span class="st">'biqu'</span><span class="op">)</span></span>
<span>        <span class="op">}</span></span>
<span>    <span class="op">)</span>,</span>
<span>    <span class="va">keys_ubi</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ubi</span></span></code></pre></div>
<pre><code><span><span class="co">## $SIGNOR_mechanism</span></span>
<span><span class="co">## [1] "ubiquitination"     "Ubiquitination"     "deubiquitination"  </span></span>
<span><span class="co">## [4] "polyubiquitination" "monoubiquitination"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`HPRD-phos_mechanism`</span></span>
<span><span class="co">## [1] "Ubiquitination"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $SPIKE_mechanism</span></span>
<span><span class="co">## [1] "Ubiquitination"     "Polyubiquitination"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $SPIKE_LC_mechanism</span></span>
<span><span class="co">## [1] "Ubiquitination"     "Polyubiquitination"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $CA1_type</span></span>
<span><span class="co">## [1] "Ubiquitination"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Macrophage_type</span></span>
<span><span class="co">## [1] "Ubiquitination"</span></span></code></pre>
<p>Actually to match all ubiquitination interactions, it’s enough to
filter for “ubiquitination” in its lowercase and capitalized forms
(note, we could also include deubiqutination and
polyubiquitination):</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ubi_kws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ubiquitination'</span>, <span class="st">'Ubiquitination'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">i_ubi</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span></span>
<span>            <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>                <span class="va">keys_ubi</span>,</span>
<span>                <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span><span class="op">{</span></span>
<span>                    <span class="fu"><a href="../reference/filter_extra_attrs.html">filter_extra_attrs</a></span><span class="op">(</span><span class="va">i</span>, <span class="op">!</span><span class="op">!</span><span class="va">k</span> <span class="op">:=</span> <span class="va">ubi_kws</span>, na_ok <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">i_ubi</span>, <span class="va">source_genesymbol</span>, <span class="va">target_genesymbol</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 49,308 × 2</span></span></span>
<span><span class="co">##    source_genesymbol target_genesymbol</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> NUMB              NOTCH1           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> BTRC_CUL1_SKP1    PER2             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> PRKN              RANBP2           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> PRKN              SNCA             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> FBXW7             MYC              </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> UBE2T             FANCL            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> BIRC2             TRAF2            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> TRAF2             MAP3K14          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> TRAF6             MAP3K7           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> BTRC_CUL1_SKP1    WEE1             </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 49,298 more rows</span></span></span></code></pre>
<p>We found 405 ubiquitination interactions. We had to use
<code>map</code>, <code>bind_rows</code> and <code>distinct</code>
because otherwise <code>filter_extra_attrs</code> would return the
intersection of the matches, instead of their union.</p>
<p>In this data frame we have 150 unique ubiquitin E3 ligases:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">i_ubi</span><span class="op">$</span><span class="va">source_genesymbol</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 365</span></span></code></pre>
<p>UniProt annotates E3 ligases by the “Ubl conjugation” keyword. We can
check how many of those 150 proteins have this annotation:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">uniprot_kws</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span></span>
<span>    resources <span class="op">=</span> <span class="st">'UniProt_keyword'</span>,</span>
<span>    entity_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>    wide <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">e3_ligases</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">uniprot_kws</span>, <span class="va">keyword</span> <span class="op">==</span> <span class="st">'Ubl conjugation'</span><span class="op">)</span>,</span>
<span>    <span class="va">genesymbol</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">e3_ligases</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2542</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">i_ubi</span><span class="op">$</span><span class="va">source_genesymbol</span><span class="op">)</span>, <span class="va">e3_ligases</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 106</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">i_ubi</span><span class="op">$</span><span class="va">source_genesymbol</span><span class="op">)</span>, <span class="va">e3_ligases</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 259</span></span></code></pre>
<p>We retrieved 2503 E3 ligases from UniProt. 83 of these has substrates
in the interaction database, while 67 of the effectors of the
interactions are not annotated in UniProt.</p>
<p>In the OmniPath enzyme-substrate database we collect ubiquitination
interactions from enzyme-PTM resources. However, these contain only a
small number of interactions:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">es_ubi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span><span class="op">(</span>types <span class="op">=</span> <span class="st">'ubiquitination'</span><span class="op">)</span></span>
<span><span class="va">es_ubi</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 70 × 12</span></span></span>
<span><span class="co">##    enzyme substrate enzyme_genesymbol substrate_genesymbol residue_type</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Q12933 Q13546    TRAF2             RIPK1                K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Q8IUD6 O95786    RNF135            RIGI                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Q8IUD6 O95786    RNF135            RIGI                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> P60604 Q92813    UBE2G2            DIO2                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> P60604 Q92813    UBE2G2            DIO2                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Q13489 Q13546    BIRC3             RIPK1                K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Q96J02 Q7Z434    ITCH              MAVS                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Q96J02 Q7Z434    ITCH              MAVS                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Q66K89 P04637    E4F1              TP53                 K           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Q66K89 P04637    E4F1              TP53                 K           </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 60 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 7 more variables: residue_offset &lt;dbl&gt;, modification &lt;chr&gt;, sources &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   references &lt;chr&gt;, curation_effort &lt;dbl&gt;, n_references &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   n_resources &lt;int&gt;</span></span></span></code></pre>
<p>With only two exception, all these have been recovered by using the
extra attributes from the network database:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">es_i_ubi</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span></span>
<span>        <span class="va">es_ubi</span>,</span>
<span>        <span class="va">i_ubi</span>,</span>
<span>        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">'enzyme_genesymbol'</span> <span class="op">=</span> <span class="st">'source_genesymbol'</span>,</span>
<span>            <span class="st">'substrate_genesymbol'</span> <span class="op">=</span> <span class="st">'target_genesymbol'</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">es_i_ubi</span>, <span class="va">enzyme</span>, <span class="va">substrate</span>, <span class="va">residue_offset</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 57</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Arch Linux</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/libopenblas.so.0.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/liblapack.so.3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Berlin</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] OmnipathR_3.9.6  BiocStyle_2.28.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] tidyr_1.3.0         rappdirs_0.3.3      sass_0.4.6         </span></span>
<span><span class="co">##  [4] utf8_1.2.3          generics_0.1.3      xml2_1.3.4         </span></span>
<span><span class="co">##  [7] stringi_1.7.12      hms_1.1.3           digest_0.6.31      </span></span>
<span><span class="co">## [10] magrittr_2.0.3      evaluate_0.21       bookdown_0.34      </span></span>
<span><span class="co">## [13] fastmap_1.1.1       cellranger_1.1.0    rprojroot_2.0.3    </span></span>
<span><span class="co">## [16] jsonlite_1.8.4      progress_1.2.2      backports_1.4.1    </span></span>
<span><span class="co">## [19] BiocManager_1.30.20 rvest_1.0.3         httr_1.4.6         </span></span>
<span><span class="co">## [22] purrr_1.0.1         fansi_1.0.4         textshaping_0.3.6  </span></span>
<span><span class="co">## [25] jquerylib_0.1.4     cli_3.6.1           rlang_1.1.1        </span></span>
<span><span class="co">## [28] crayon_1.5.2        bit64_4.0.5         withr_2.5.0        </span></span>
<span><span class="co">## [31] cachem_1.0.8        yaml_2.3.7          parallel_4.3.0     </span></span>
<span><span class="co">## [34] tools_4.3.0         tzdb_0.4.0          memoise_2.0.1      </span></span>
<span><span class="co">## [37] checkmate_2.2.0     dplyr_1.1.2         curl_5.0.0         </span></span>
<span><span class="co">## [40] vctrs_0.6.2         logger_0.2.2        R6_2.5.1           </span></span>
<span><span class="co">## [43] lifecycle_1.0.3     stringr_1.5.0       bit_4.0.5          </span></span>
<span><span class="co">## [46] fs_1.6.2            vroom_1.6.3         ragg_1.2.5         </span></span>
<span><span class="co">## [49] pkgconfig_2.0.3     desc_1.4.2          pkgdown_2.0.7      </span></span>
<span><span class="co">## [52] bslib_0.4.2         pillar_1.9.0        later_1.3.1        </span></span>
<span><span class="co">## [55] glue_1.6.2          Rcpp_1.0.10         systemfonts_1.0.4  </span></span>
<span><span class="co">## [58] xfun_0.39           tibble_3.2.1        tidyselect_1.2.0   </span></span>
<span><span class="co">## [61] knitr_1.43          htmltools_0.5.5     igraph_1.4.3       </span></span>
<span><span class="co">## [64] rmarkdown_2.22      readr_2.1.4         compiler_4.3.0     </span></span>
<span><span class="co">## [67] prettyunits_1.1.1   readxl_1.4.2</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
