<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="OmnipathR">
<title>OmnipathR: an R client for the OmniPath web service • OmnipathR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="OmnipathR: an R client for the OmniPath web service">
<meta property="og:description" content="OmnipathR">
<meta property="og:image" content="https://saezlab.github.io/OmnipathR/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-10"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-10');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">OmnipathR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.13.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bioc_workshop.html">OmniPath Bioconductor workshop</a>
    <a class="dropdown-item" href="../articles/cosmos.html">Building prior knowledge network (PKN) for COSMOS</a>
    <a class="dropdown-item" href="../articles/db_manager.html">The database manager in OmnipathR</a>
    <a class="dropdown-item" href="../articles/drug_targets.html">Building protein networks around drug-targets using OmnipathR</a>
    <a class="dropdown-item" href="../articles/extra_attrs.html">Resource specific interaction attributes</a>
    <a class="dropdown-item" href="../articles/gene_ontology.html">Gene Ontology in OmnipathR</a>
    <a class="dropdown-item" href="../articles/nichenet.html">Using NicheNet with OmnipathR</a>
    <a class="dropdown-item" href="../articles/omnipath_intro.html">OmnipathR: an R client for the OmniPath web service</a>
    <a class="dropdown-item" href="../articles/paths.html">Pathway construction</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/OmnipathR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/omnipathdb">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://omnipathdb.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>OmnipathR: an R client for the OmniPath web service</h1>
                        <h4 data-toc-skip class="author">Alberto Valdeolivas</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:alvaldeolivas@gmail.com" class="email">alvaldeolivas@gmail.com</a>
      
                              <h4 data-toc-skip class="author">Attila Gabor</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:attila.gabor@bioquant.uni-heidelberg.de" class="email">attila.gabor@bioquant.uni-heidelberg.de</a>
      
                              <h4 data-toc-skip class="author">Denes Turei</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:turei.denes@gmail.com" class="email">turei.denes@gmail.com</a>
      
                              <h4 data-toc-skip class="author">Julio Saez-Rodriguez</h4>
            <address class="author_afil">
      Institute for Computational Biomedicine, Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/OmnipathR/blob/HEAD/vignettes/omnipath_intro.Rmd" class="external-link"><code>vignettes/omnipath_intro.Rmd</code></a></small>
      <div class="d-none name"><code>omnipath_intro.Rmd</code></div>
    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to use the <em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em>
      package to retrieve information from the OmniPath database:
      <a href="https://omnipathdb.org/" class="external-link uri">https://omnipathdb.org/</a>.
      In addition, it includes some utility functions to filter
      analyse and visualize the data.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em> is an <em>R</em> package built to provide easy access to
the data stored in the OmniPath webservice <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>:</p>
<p><a href="http://omnipathdb.org/" class="external-link uri">http://omnipathdb.org/</a></p>
<p>The webservice implements a very simple REST style API. This package make
requests by the HTTP protocol to retreive the data. Hence, fast Internet
access is required for a proper use of <em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em>.</p>
<div class="section level3">
<h3 id="query-types">Query types<a class="anchor" aria-label="anchor" href="#query-types"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em> can retrieve five different types of data:</p>
<ul>
<li>
<p><strong>Interactions</strong>: protein-protein interactions organized in different
datasets:</p>
<ul>
<li>
<strong>omnipath</strong>: the OmniPath data as defined in the original
publication <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span> and collected from different databases.</li>
<li>
<strong>pathwayextra</strong>: activity flow interactions without literature
reference.</li>
<li>
<strong>kinaseextra</strong>: enzyme-substrate interactions without literature
reference.</li>
<li>
<strong>ligrecextra</strong>: ligand-receptor interactions without literature
reference.</li>
<li>
<strong>tfregulons</strong>: transcription factor (TF)-target interactions from
<strong>DoRothEA</strong> <span class="citation">(Garcia-Alonso et al. 2019)</span>.</li>
<li>
<strong>tf-miRNA</strong>: transcription factor-miRNA interactions</li>
<li>
<strong>miRNA-target</strong>: miRNA-mRNA interactions.</li>
<li>
<strong>lncRNA-mRNA</strong>: lncRNA-mRNA interactions.</li>
<li>
<strong>small molecule-protein</strong>: interactions between small molecules
(metabolites, intrinsic ligands, drug compounds) and proteins.</li>
</ul>
</li>
<li><p><strong>Post-translational modifications (PTMs)</strong>: It provides
enzyme-substrate reactions in a very similar way to the aforementioned
interactions. Some of the biological databases related to PTMs integrated
in OmniPath are Phospho.ELM <span class="citation">(Dinkel et al. 2010)</span> and PhosphoSitePlus
[<span class="citation">Hornbeck et al. (2014)</span>}.</p></li>
<li><p><strong>Complexes</strong>: it provides access to a comprehensive database of more than
22000 protein complexes. This data comes from different resources
such as: CORUM <span class="citation">(Giurgiu et al. 2018)</span> or Hu.map <span class="citation">(Drew et al. 2017)</span>.</p></li>
<li><p><strong>Annotations</strong>: it provides a large variety of data regarding
different annotations about proteins and complexes. These data come from
dozens of databases covering different topics such as: The Topology Data Bank
of Transmembrane Proteins (TOPDB) <span class="citation">(Dobson et al. 2014)</span> or ExoCarta
<span class="citation">(Keerthikumar et al. 2016)</span>, a database collecting the proteins that were identified
in exosomes in multiple organisms.</p></li>
<li><p><strong>Intercell</strong>: it provides information on the roles in
inter-cellular signaling. For instance. if a protein is a ligand, a receptor,
an extracellular matrix (ECM) component, etc. The data does not come from
original sources but combined from several databases by us. The source
databases, such as CellPhoneDB <span class="citation">(Vento-Tormo et al. 2018)</span> or Receptome
<span class="citation">(Ben-Shlomo et al. 2003)</span>, are also referred for each reacord.</p></li>
</ul>
<p>Figure <a href="#fig:fig1">1</a> shows an overview of the resources featured in
<strong>OmniPath</strong>. For more detailed information about the original data sources
integrated in Omnipath, please visit:</p>
<ul>
<li><a href="http://omnipathdb.org/" class="external-link uri">http://omnipathdb.org/</a></li>
<li><a href="http://omnipathdb.org/info" class="external-link uri">http://omnipathdb.org/info</a></li>
</ul>
<div class="figure">
<span style="display:block;" id="fig:fig1"></span>
<img src="reference/figures/page1_1.png" alt="Overview of the resources featured in OmniPath. Causal resources (including activity-flow and enzyme-substrate resources) can provide direction (*) or sign and direction (+) of interactions." width="1500"><p class="caption">
Figure 1: Overview of the resources featured in OmniPath. Causal resources (including activity-flow and enzyme-substrate resources) can provide direction (*) or sign and direction (+) of interactions.
</p>
</div>
</div>
<div class="section level3">
<h3 id="mouse-and-rat">Mouse and rat<a class="anchor" aria-label="anchor" href="#mouse-and-rat"></a>
</h3>
<p>Excluding the miRNA interactions, all interactions and PTMs are available for
human, mouse and rat. The rodent data has been translated from human using the
NCBI Homologene database. Many human proteins do not have known homolog in
rodents, hence rodent datasets are smaller than their human counterparts.</p>
<p>In case you work with mouse omics data you might do better to translate your
dataset to human (for example using the pypath.homology module,
<a href="https://github.com/saezlab/pypath/" class="external-link uri">https://github.com/saezlab/pypath/</a>) and use human interaction data.</p>
</div>
</div>
<div class="section level2">
<h2 id="installation-of-the-omnipathr-package">Installation of the <em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em> package<a class="anchor" aria-label="anchor" href="#installation-of-the-omnipathr-package"></a>
</h2>
<p>First of all, you need a current version of <em>R</em>. <em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em>
is a freely available package deposited on <em>Bioconductor</em> and
<a href="https://github.com/saezlab/OmnipathR" class="external-link">GitHub</a>. You can install it by running
the following commands on an <em>R</em> console:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"OmnipathR"</span><span class="op">)</span></span></code></pre></div>
<p>We also load here the required packages to run the code in this vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.omnipathdb.org/" class="external-link">OmnipathR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage-examples">Usage Examples<a class="anchor" aria-label="anchor" href="#usage-examples"></a>
</h2>
<p>In the following paragraphs, we provide some examples to describe how to use
the <em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em> package to retrieve different types of
information from <strong>Omnipath</strong> webserver. In addition, we play around with the
data aiming at obtaining some biological relevant information.</p>
<p>Noteworthy, the sections <strong>complexes</strong>, <strong>annotations</strong> and <strong>intercell</strong> are
linked. We explore the annotations and roles in inter-cellular communications
of the proteins involved in a given complex. This basic example shows the
usefulness of integrating the information available in the different
<strong>Omnipath</strong> resources.</p>
<div class="section level3">
<h3 id="interactions">Interactions<a class="anchor" aria-label="anchor" href="#interactions"></a>
</h3>
<p>Proteins interact among them and with other biological molecules to perform
cellular functions. Proteins also participates in pathways, linked series of
reactions occurring inter/intra cells to transform products or to transmit
signals inducing specific cellular responses. Protein interactions are
therefore a very valuable source of information to understand cellular
functioning.</p>
<p>We here download the original <strong>OmniPath</strong> human interactions <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>.
To do so, we first check the different source databases and select some of
them. Then, we print some of the downloaded interactions (“+” means
activation, “-” means inhibition and “?” means undirected interactions or
inconclusive data).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We check some of the different interaction databases</span></span>
<span><span class="fu"><a href="../reference/get_interaction_resources.html">get_interaction_resources</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   [1] "ABS"                         "ACSN"                       </span></span>
<span><span class="co">##   [3] "ACSN_SignaLink3"             "Adhesome"                   </span></span>
<span><span class="co">##   [5] "AlzPathway"                  "ARACNe-GTEx_DoRothEA"       </span></span>
<span><span class="co">##   [7] "ARN"                         "Baccin2019"                 </span></span>
<span><span class="co">##   [9] "BEL-Large-Corpus_ProtMapper" "BioGRID"                    </span></span>
<span><span class="co">##  [11] "CA1"                         "CancerCellMap"              </span></span>
<span><span class="co">##  [13] "CancerDrugsDB"               "CellCall"                   </span></span>
<span><span class="co">##  [15] "CellChatDB"                  "CellChatDB-cofactors"       </span></span>
<span><span class="co">##  [17] "Cellinker"                   "CellPhoneDB"                </span></span>
<span><span class="co">##  [19] "CellPhoneDB_Cellinker"       "CellTalkDB"                 </span></span>
<span><span class="co">##  [21] "CollecTRI"                   "connectomeDB2020"           </span></span>
<span><span class="co">##  [23] "Cui2007"                     "CytReg_CollecTRI"           </span></span>
<span><span class="co">##  [25] "dbPTM"                       "DeathDomain"                </span></span>
<span><span class="co">##  [27] "DEPOD"                       "DIP"                        </span></span>
<span><span class="co">##  [29] "DLRP_Cellinker"              "DLRP_talklr"                </span></span>
<span><span class="co">##  [31] "DOMINO"                      "DoRothEA"                   </span></span>
<span><span class="co">##  [33] "DoRothEA-A_CollecTRI"        "DoRothEA-reviews_DoRothEA"  </span></span>
<span><span class="co">##  [35] "ELM"                         "EMBRACE"                    </span></span>
<span><span class="co">##  [37] "ENCODE_tf-mirna"             "ENCODE-distal"              </span></span>
<span><span class="co">##  [39] "ENCODE-proximal"             "ExTRI_CollecTRI"            </span></span>
<span><span class="co">##  [41] "FANTOM4_DoRothEA"            "Fantom5_LRdb"               </span></span>
<span><span class="co">##  [43] "GEREDB_CollecTRI"            "GOA_CollecTRI"              </span></span>
<span><span class="co">##  [45] "Guide2Pharma"                "Guide2Pharma_Cellinker"     </span></span>
<span><span class="co">##  [47] "Guide2Pharma_LRdb"           "Guide2Pharma_talklr"        </span></span>
<span><span class="co">##  [49] "HOCOMOCO_DoRothEA"           "HPMR"                       </span></span>
<span><span class="co">##  [51] "HPMR_Cellinker"              "HPMR_LRdb"                  </span></span>
<span><span class="co">##  [53] "HPMR_talklr"                 "HPRD"                       </span></span>
<span><span class="co">##  [55] "HPRD_KEA"                    "HPRD_LRdb"                  </span></span>
<span><span class="co">##  [57] "HPRD_MIMP"                   "HPRD_talklr"                </span></span>
<span><span class="co">##  [59] "HPRD-phos"                   "HTRI_CollecTRI"             </span></span>
<span><span class="co">##  [61] "HTRIdb"                      "HTRIdb_DoRothEA"            </span></span>
<span><span class="co">##  [63] "HuRI"                        "ICELLNET"                   </span></span>
<span><span class="co">##  [65] "InnateDB"                    "InnateDB_SignaLink3"        </span></span>
<span><span class="co">##  [67] "IntAct"                      "IntAct_CollecTRI"           </span></span>
<span><span class="co">##  [69] "IntAct_DoRothEA"             "iPTMnet"                    </span></span>
<span><span class="co">##  [71] "iTALK"                       "JASPAR_DoRothEA"            </span></span>
<span><span class="co">##  [73] "KEA"                         "KEGG-MEDICUS"               </span></span>
<span><span class="co">##  [75] "Kinexus_KEA"                 "Kirouac2010"                </span></span>
<span><span class="co">##  [77] "Li2012"                      "Lit-BM-17"                  </span></span>
<span><span class="co">##  [79] "LncRNADisease"               "LRdb"                       </span></span>
<span><span class="co">##  [81] "Macrophage"                  "MatrixDB"                   </span></span>
<span><span class="co">##  [83] "MIMP"                        "miR2Disease"                </span></span>
<span><span class="co">##  [85] "miRDeathDB"                  "miRecords"                  </span></span>
<span><span class="co">##  [87] "miRTarBase"                  "MPPI"                       </span></span>
<span><span class="co">##  [89] "NCI-PID_ProtMapper"          "ncRDeathDB"                 </span></span>
<span><span class="co">##  [91] "NetPath"                     "NetworKIN_KEA"              </span></span>
<span><span class="co">##  [93] "NFIRegulomeDB_DoRothEA"      "NRF2ome"                    </span></span>
<span><span class="co">##  [95] "NTNU.Curated_CollecTRI"      "ORegAnno"                   </span></span>
<span><span class="co">##  [97] "ORegAnno_DoRothEA"           "Pavlidis2021_CollecTRI"     </span></span>
<span><span class="co">##  [99] "PAZAR_DoRothEA"              "phosphoELM"                 </span></span>
<span><span class="co">## [101] "phosphoELM_KEA"              "phosphoELM_MIMP"            </span></span>
<span><span class="co">## [103] "PhosphoNetworks"             "PhosphoPoint"               </span></span>
<span><span class="co">## [105] "PhosphoSite"                 "PhosphoSite_KEA"            </span></span>
<span><span class="co">## [107] "PhosphoSite_MIMP"            "PhosphoSite_noref"          </span></span>
<span><span class="co">## [109] "PhosphoSite_ProtMapper"      "ProtMapper"                 </span></span>
<span><span class="co">## [111] "Ramilowski2015"              "Ramilowski2015_Baccin2019"  </span></span>
<span><span class="co">## [113] "REACH_ProtMapper"            "Reactome_LRdb"              </span></span>
<span><span class="co">## [115] "Reactome_ProtMapper"         "Reactome_SignaLink3"        </span></span>
<span><span class="co">## [117] "RegNetwork_DoRothEA"         "ReMap_DoRothEA"             </span></span>
<span><span class="co">## [119] "RLIMS-P_ProtMapper"          "scConnect"                  </span></span>
<span><span class="co">## [121] "SignaLink3"                  "SIGNOR"                     </span></span>
<span><span class="co">## [123] "SIGNOR_CollecTRI"            "SIGNOR_ProtMapper"          </span></span>
<span><span class="co">## [125] "Sparser_ProtMapper"          "SPIKE"                      </span></span>
<span><span class="co">## [127] "SPIKE_LC"                    "STRING_talklr"              </span></span>
<span><span class="co">## [129] "talklr"                      "TCRcuration_SignaLink3"     </span></span>
<span><span class="co">## [131] "TFactS_CollecTRI"            "TFactS_DoRothEA"            </span></span>
<span><span class="co">## [133] "TFe_DoRothEA"                "TransmiR"                   </span></span>
<span><span class="co">## [135] "TRED_DoRothEA"               "TRIP"                       </span></span>
<span><span class="co">## [137] "TRRD_DoRothEA"               "TRRUST_CollecTRI"           </span></span>
<span><span class="co">## [139] "TRRUST_DoRothEA"             "UniProt_LRdb"               </span></span>
<span><span class="co">## [141] "Wang"                        "Wojtowicz2020"</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The interactions are stored into a data frame.</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span><span class="op">(</span></span>
<span>        resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SignaLink3"</span>, <span class="st">"PhosphoSite"</span>, <span class="st">"SIGNOR"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co">## We visualize the first interactions in the data frame.</span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">##   source          interaction target         n_resources n_references</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> SRC (P12931)    ==( + )==&gt;  TRPV1 (Q8NER1)           5            6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> PRKACA (P17612) ==( ? )==&gt;  TRPC6 (Q9Y210)           2            3</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> PRKG1 (Q13976)  ==( - )==&gt;  TRPC3 (Q13507)           8            2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> PTPN1 (P18031)  ==( - )==&gt;  TRPV6 (Q9H1D0)           6            2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> PRKG1 (Q13976)  ==( + )==&gt;  TRPC7 (Q9HCX4)           3            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> OS9 (Q13438)    ==(+/-)==&gt;  TRPV4 (Q9HBA0)           3            1</span></span></code></pre>
<div class="section level4">
<h4 id="protein-protein-interaction-networks">Protein-protein interaction networks<a class="anchor" aria-label="anchor" href="#protein-protein-interaction-networks"></a>
</h4>
<p>Protein-protein interactions are usually converted into networks. Describing
protein interactions as networks not only provides a convenient format for
visualization, but also allows applying graph theory methods to mine the
biological information they contain.</p>
<p>We convert here our set of interactions to a network/graph
(<em><a href="https://CRAN.R-project.org/package=igraph" class="external-link">igraph</a></em>object). Then, we apply two very common approaches to
extract information from a biological network:</p>
<ul>
<li>
<strong>Shortest Paths</strong>: finding a path between two nodes (proteins) going
through the minimum number of edges. This can be very useful to track
consecutive reactions within a given pathway. We display below the shortest
path between two given proteins and all the possible shortests paths between
two other proteins. It is to note that the functions <code>print_path\_es</code> and
<code>print_path\_vs</code> display very similar results, but the first one takes as an
input an edge sequence and the second one a node sequence.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We transform the interactions data frame into a graph</span></span>
<span><span class="va">pathways_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find and print shortest paths on the directed network between proteins</span></span>
<span><span class="co">## of interest:</span></span>
<span><span class="fu"><a href="../reference/print_path_es.html">print_path_es</a></span><span class="op">(</span></span>
<span>    <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/distances.html" class="external-link">shortest_paths</a></span><span class="op">(</span></span>
<span>        <span class="va">pathways_g</span>,</span>
<span>        from <span class="op">=</span> <span class="st">"TYRO3"</span>,</span>
<span>        to <span class="op">=</span> <span class="st">"STAT3"</span>,</span>
<span>        output <span class="op">=</span> <span class="st">"epath"</span></span>
<span>    <span class="op">)</span><span class="op">$</span><span class="va">epath</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    <span class="va">pathways_g</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            source interaction          target n_resources n_references</span></span>
<span><span class="co">## 1  TYRO3 (Q06418)  ==( + )==&gt; PIK3R1 (P27986)           4            3</span></span>
<span><span class="co">## 2 PIK3R1 (P27986)  ==(+/-)==&gt; PIK3CG (P48736)           4            3</span></span>
<span><span class="co">## 3 PIK3CG (P48736)  ==( + )==&gt;   RAC1 (P63000)           2            2</span></span>
<span><span class="co">## 4   RAC1 (P63000)  ==( + )==&gt;  STAT3 (P40763)           9            4</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Find and print all shortest paths between proteins of interest:</span></span>
<span><span class="fu"><a href="../reference/print_path_vs.html">print_path_vs</a></span><span class="op">(</span></span>
<span>    <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/distances.html" class="external-link">all_shortest_paths</a></span><span class="op">(</span></span>
<span>        <span class="va">pathways_g</span>,</span>
<span>        from <span class="op">=</span> <span class="st">"DYRK2"</span>,</span>
<span>        to <span class="op">=</span> <span class="st">"MAPKAPK2"</span></span>
<span>    <span class="op">)</span><span class="op">$</span><span class="va">res</span>,</span>
<span>    <span class="va">pathways_g</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Pathway 1: DYRK2 -&gt; NOTCH1 -&gt; RBPJ -&gt; DUSP1 -&gt; MAPK3 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 2: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PEA15 -&gt; MAPK3 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 3: DYRK2 -&gt; NOTCH1 -&gt; RBPJ -&gt; DUSP1 -&gt; MAPK14 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 4: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; GSK3B -&gt; MAPK14 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 5: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP2K4 -&gt; MAPK14 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 6: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; MAP3K5 -&gt; MAPK14 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 7: DYRK2 -&gt; NOTCH1 -&gt; RBPJ -&gt; DUSP1 -&gt; MAPK1 -&gt; MAPKAPK2</span></span></code></pre>
<pre><code><span><span class="co">## Pathway 8: DYRK2 -&gt; TBK1 -&gt; AKT1 -&gt; PEA15 -&gt; MAPK1 -&gt; MAPKAPK2</span></span></code></pre>
<ul>
<li>
<strong>Clustering</strong>: grouping nodes (proteins) in such a way that nodes belonging
to the same group (called cluster) are more connected in the network to each
other than to those in other groups (clusters). Since proteins interact to
perform their functions, proteins within the same cluster are likely to be
implicated in similar biological tasks. Figure <a href="#fig:fig2">2</a> shows the
subgraph containing the proteins and interactions of a specifc protein, ERBB2
The <em><a href="https://CRAN.R-project.org/package=igraph" class="external-link">igraph</a></em> package contains functions to apply sevaral
different cluster methods on graphs (visit <a href="https://igraph.org/r/doc/" class="external-link uri">https://igraph.org/r/doc/</a> for
detailed information.)</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We apply a clustering algorithm (Louvain) to group proteins in</span></span>
<span><span class="co">## our network. We apply here Louvain which is fast but can only run</span></span>
<span><span class="co">## on undirected graphs. Other clustering algorithms can deal with</span></span>
<span><span class="co">## directed networks but with longer computational times,</span></span>
<span><span class="co">## such as cluster_edge_betweenness. These cluster methods are directly</span></span>
<span><span class="co">## available in the igraph package.</span></span>
<span><span class="va">pathways_g_u</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/as.directed.html" class="external-link">as.undirected</a></span><span class="op">(</span><span class="va">pathways_g</span>, mode <span class="op">=</span> <span class="st">"mutual"</span><span class="op">)</span></span>
<span><span class="va">pathways_g_u</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/simplify.html" class="external-link">simplify</a></span><span class="op">(</span><span class="va">pathways_g_u</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/cluster_fast_greedy.html" class="external-link">cluster_fast_greedy</a></span><span class="op">(</span><span class="va">pathways_g_u</span><span class="op">)</span></span>
<span><span class="co">## We extract the cluster where a protein of interest is contained</span></span>
<span><span class="va">erbb2_cluster_id</span> <span class="op">&lt;-</span> <span class="va">clusters</span><span class="op">$</span><span class="va">membership</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">$</span><span class="va">names</span> <span class="op">==</span> <span class="st">"ERBB2"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">erbb2_cluster_g</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/subgraph.html" class="external-link">induced_subgraph</a></span><span class="op">(</span></span>
<span>    <span class="va">pathways_g_u</span>,</span>
<span>    <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">pathways_g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">$</span><span class="va">membership</span> <span class="op">==</span> <span class="va">erbb2_cluster_id</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:fig2"></span>
<img src="omnipath_intro_files/figure-html/fig2-1.png" alt="ERBB2 associated cluser. Subnetwork extracted from the interactions graph representing the cluster where we can find the gene *ERBB2* (yellow node)"><p class="caption">
Figure 2: ERBB2 associated cluser. Subnetwork extracted from the interactions graph representing the cluster where we can find the gene <em>ERBB2</em> (yellow node)
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="other-interaction-datasets">Other interaction datasets<a class="anchor" aria-label="anchor" href="#other-interaction-datasets"></a>
</h3>
<p>We used above the interactions from the dataset described in the original
<strong>OmniPath</strong> publication <span class="citation">(Türei, Korcsmáros, and Saez-Rodriguez 2016)</span>. In this section, we provide
examples on how to retry and deal with interactions from the remaining
datasets. The same functions can been applied to every interaction dataset.</p>
<div class="section level4">
<h4 id="pathway-extra">Pathway Extra<a class="anchor" aria-label="anchor" href="#pathway-extra"></a>
</h4>
<p>In the first example, we are going to get the interactions from the
<strong>pathwayextra</strong> dataset, which contains activity flow interactions
without literature reference. We are going to focus on the mouse interactions
for a given gene in this particular case.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store the interactions into a dataframe</span></span>
<span><span class="va">iactions</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="../reference/import_pathwayextra_interactions.html">import_pathwayextra_interactions</a></span><span class="op">(</span></span>
<span>        resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Wang"</span>, <span class="st">"Lit-BM-17"</span><span class="op">)</span>,</span>
<span>        organism <span class="op">=</span> <span class="fl">10090</span>  <span class="co"># mouse</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co">## We select all the interactions in which Amfr gene is involved</span></span>
<span><span class="va">iactions_amfr</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">iactions</span>,</span>
<span>    <span class="va">source_genesymbol</span> <span class="op">==</span> <span class="st">"Amfr"</span> <span class="op">|</span></span>
<span>    <span class="va">target_genesymbol</span> <span class="op">==</span> <span class="st">"Amfr"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We print these interactions:</span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">iactions_amfr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 5</span></span></span>
<span><span class="co">##   source        interaction target        n_resources n_references</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Amfr (Q9R049) ==( + )==&gt;  Vcp (Q01853)            6           20</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Gpi (P06745)  ==( + )==&gt;  Amfr (Q9R049)          13            4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Amfr (Q9R049) ==( - )==&gt;  Sod1 (P08228)           1            0</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="kinase-extra">Kinase Extra<a class="anchor" aria-label="anchor" href="#kinase-extra"></a>
</h4>
<p>Next, we download the interactions from the <strong>kinaseextra</strong> dataset, which
contains enzyme-substrate interactions without literature reference. We are
going to focus on rat reactions targeting a particular gene.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store the interactions into a dataframe</span></span>
<span><span class="va">phosphonetw</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="../reference/import_kinaseextra_interactions.html">import_kinaseextra_interactions</a></span><span class="op">(</span></span>
<span>        resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PhosphoPoint"</span>, <span class="st">"PhosphoSite"</span><span class="op">)</span>,</span>
<span>        organism <span class="op">=</span> <span class="fl">10116</span>  <span class="co"># rat</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co">## We select the interactions in which Dpysl2 gene is a target</span></span>
<span><span class="va">upstream_dpysl2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">phosphonetw</span>,</span>
<span>    <span class="va">target_genesymbol</span> <span class="op">==</span> <span class="st">"Dpysl2"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We print these interactions:</span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">upstream_dpysl2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 5 × 5</span></span></span>
<span><span class="co">##   source         interaction target          n_resources n_references</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Gsk3b (P18266) ==(+/-)==&gt;  Dpysl2 (P47942)          12           33</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Cdk5 (Q03114)  ==(+/-)==&gt;  Dpysl2 (P47942)           6           30</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Rock2 (Q62868) ==( + )==&gt;  Dpysl2 (P47942)          11            6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Rock1 (Q63644) ==( ? )==&gt;  Dpysl2 (P47942)           7            2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Fer (P09760)   ==( ? )==&gt;  Dpysl2 (P47942)           2            2</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="ligand-receptor-extra">Ligand-receptor Extra<a class="anchor" aria-label="anchor" href="#ligand-receptor-extra"></a>
</h4>
<p>In the following example we are going to work with the <strong>ligrecextra</strong>
dataset, which contains ligand-receptor interactions without literature
reference. Our goal is to find the potential receptors associated to a given
ligand, CDH1 (Figure <a href="#fig:fig3">3</a>).</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store the interactions into a dataframe</span></span>
<span><span class="va">ligrec_netw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_ligrecextra_interactions.html">import_ligrecextra_interactions</a></span><span class="op">(</span></span>
<span>    resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"iTALK"</span>, <span class="st">"Baccin2019"</span><span class="op">)</span>,</span>
<span>    organism <span class="op">=</span> <span class="fl">9606</span>  <span class="co"># human</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Receptors of the CDH1 ligand.</span></span>
<span><span class="va">downstream_cdh1</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">ligrec_netw</span>,</span>
<span>    <span class="va">source_genesymbol</span> <span class="op">==</span> <span class="st">"CDH1"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We transform the interactions data frame into a graph</span></span>
<span><span class="va">downstream_cdh1_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span><span class="va">downstream_cdh1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We induce a network with these genes</span></span>
<span><span class="va">downstream_cdh1_g</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/subgraph.html" class="external-link">induced_subgraph</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span><span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">downstream_cdh1_g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:fig3"></span>
<img src="omnipath_intro_files/figure-html/fig3-1.png" alt="Ligand-receptor interactions for the ADM2 ligand."><p class="caption">
Figure 3: Ligand-receptor interactions for the ADM2 ligand.
</p>
</div>
</div>
<div class="section level4">
<h4 id="dorothea-regulons">DoRothEA Regulons<a class="anchor" aria-label="anchor" href="#dorothea-regulons"></a>
</h4>
<p>Another very interesting interaction dataset also available in OmniPath is
<strong>DoRothEA</strong> <span class="citation">(Garcia-Alonso et al. 2019)</span>. It contains transcription
factor (TF)-target interactions with confidence score, ranging from A-E,
being A the most confident interactions. In the code chunk shown below, we
select and print the most confident interactions for a given TF.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store the interactions into a dataframe</span></span>
<span><span class="va">dorothea_netw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dorothea.html">dorothea</a></span><span class="op">(</span></span>
<span>    dorothea_levels <span class="op">=</span> <span class="st">"A"</span>,</span>
<span>    organism <span class="op">=</span> <span class="fl">9606</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We select the most confident interactions for a given TF and we print</span></span>
<span><span class="co">## the interactions to check the way it regulates its different targets</span></span>
<span><span class="va">downstream_gli1</span>  <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">dorothea_netw</span>,</span>
<span>    <span class="va">source_genesymbol</span> <span class="op">==</span> <span class="st">"GLI1"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">downstream_gli1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 7 × 5</span></span></span>
<span><span class="co">##   source        interaction target          n_resources n_references</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> GLI1 (P08151) ==( + )==&gt;  IGFBP6 (P24592)           1            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> GLI1 (P08151) ==( - )==&gt;  EGR2 (P11161)             1            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> GLI1 (P08151) ==( + )==&gt;  CCND2 (P30279)            1            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> GLI1 (P08151) ==( - )==&gt;  SLIT2 (O94813)            1            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> GLI1 (P08151) ==( + )==&gt;  BCL2 (P10415)             1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> GLI1 (P08151) ==( + )==&gt;  PTCH1 (Q13635)            1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> GLI1 (P08151) ==( + )==&gt;  SFRP1 (Q8N474)            1            0</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="mirna-target-dataset">miRNA-target dataset<a class="anchor" aria-label="anchor" href="#mirna-target-dataset"></a>
</h4>
<p>The last dataset describing interactions is <strong>mirnatarget</strong>. It stores
miRNA-mRNA and TF-miRNA interactions. These interactions are only available
for human so far. We next select the miRNA interacting with the TF selected in
the previous code chunk, <em>GLI1</em>. The main function of miRNAs seems to
be related with gene regulation. It is therefore interesting to see how some
miRNA can regulate the expression of a TF which in turn regulates the
expression of other genes. Figure <a href="#fig:fig4">4</a> shows a schematic network of
the miRNA targeting <em>GLI1</em> and the genes regulated by this TF.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store the interactions into a dataframe</span></span>
<span><span class="va">mirna_target</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_mirnatarget_interactions.html">import_mirnatarget_interactions</a></span><span class="op">(</span></span>
<span>    resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"miR2Disease"</span>, <span class="st">"miRDeathDB"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We select the interactions where a miRNA is interacting with the TF</span></span>
<span><span class="co">## used in the previous code chunk and we print these interactions.</span></span>
<span><span class="va">upstream_gli1</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">mirna_target</span>,</span>
<span>    <span class="va">target_genesymbol</span> <span class="op">==</span> <span class="st">"GLI1"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">upstream_gli1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 5</span></span></span>
<span><span class="co">##   source                        interaction target      n_resources n_references</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> hsa-miR-324-5p (MIMAT0000761) ==( ? )==&gt;  GLI1 (P081…           3            2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> hsa-miR-125b (MIMAT0000423)   ==( ? )==&gt;  GLI1 (P081…           2            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> hsa-miR-326 (MIMAT0000756)    ==( ? )==&gt;  GLI1 (P081…           2            1</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We transform the previous selections to graphs (igraph objects)</span></span>
<span><span class="va">downstream_gli1_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span><span class="va">downstream_gli1</span><span class="op">)</span></span>
<span><span class="va">upstream_gli1_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interaction_graph.html">interaction_graph</a></span><span class="op">(</span><span class="va">upstream_gli1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:fig4"></span>
<img src="omnipath_intro_files/figure-html/fig4-1.png" alt="miRNA-TF-target network. Schematic network of the miRNA (red square nodes) targeting    extit{GLI1} (yellow node) and the genes regulated by this TF (blue round nodes)."><p class="caption">
Figure 4: miRNA-TF-target network. Schematic network of the miRNA (red square nodes) targeting extit{GLI1} (yellow node) and the genes regulated by this TF (blue round nodes).
</p>
</div>
</div>
<div class="section level4">
<h4 id="small-molecule-protein-dataset">Small molecule-protein dataset<a class="anchor" aria-label="anchor" href="#small-molecule-protein-dataset"></a>
</h4>
<p>This new dataset has been first added to OmniPath in January 2022. It is still
quite small: 3.5k interactions from three resources (SIGNOR, CancerDrugsDB
and Cellinker), but has prospects of a great growth in the future. As an
example, lets look for targets of a cancer drug, the MEK inhibitor Trametinib:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trametinib_targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_small_molecule_protein_interactions.html">import_small_molecule_protein_interactions</a></span><span class="op">(</span></span>
<span>    sources <span class="op">=</span> <span class="st">"TRAMETINIB"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="va">trametinib_targets</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 32 × 5</span></span></span>
<span><span class="co">##    source                interaction target          n_resources n_references</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> TRAMETINIB (11707110) ==( - )==&gt;  MAP2K1 (Q02750)           2            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> TRAMETINIB (11707110) ==( - )==&gt;  MAP2K2 (P36507)           2            1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> TRAMETINIB (11707110) ==( ? )==&gt;  MAP2K3 (P46734)           1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> TRAMETINIB (11707110) ==( ? )==&gt;  CDKN2A (Q8N726)           1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> TRAMETINIB (11707110) ==( ? )==&gt;  STAG2 (Q8N3U4)            1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> TRAMETINIB (11707110) ==( ? )==&gt;  ERBB3 (P21860)            1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> TRAMETINIB (11707110) ==( ? )==&gt;  ARAF (P10398)             1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> TRAMETINIB (11707110) ==( ? )==&gt;  DDX43 (Q9NXZ2)            1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> TRAMETINIB (11707110) ==( ? )==&gt;  GNAQ (P50148)             1            0</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> TRAMETINIB (11707110) ==( ? )==&gt;  ETV5 (P41161)             1            0</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 22 more rows</span></span></span></code></pre>
<p>Note, the human readable compound names are not reliable, use PubChem CIDs
instead.</p>
</div>
</div>
<div class="section level3">
<h3 id="post-translational-modifications-ptms">Post-translational modifications (PTMs)<a class="anchor" aria-label="anchor" href="#post-translational-modifications-ptms"></a>
</h3>
<p>Another query type available is PTMs which provides enzyme-substrate reactions
in a very similar way to the aforementioned interactions. PTMs refer
generally to enzymatic modification of proteins after their synthesis in the
ribosomes. PTMs can be highly context-specific and they play a main role
in the activation/inhibition of biological pathways.</p>
<p>In the next code chunk, we download the <strong>PTMs</strong> for human. We first check
the different available source databases, even though we do not perform any
filter. Then, we select and print the reactions involving a specific
enzyme-substrate pair. Those reactions lack information about activation or
inhibition. To obtain that information, we match the data with
<strong>OmniPath</strong> interactions. Finally, we show that it is also possible to
build a graph using this information, and to retrieve PTMs from mouse or rat.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We check the different PTMs databases</span></span>
<span><span class="fu"><a href="../reference/get_enzsub_resources.html">get_enzsub_resources</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "BEL-Large-Corpus_ProtMapper" "dbPTM"                      </span></span>
<span><span class="co">##  [3] "DEPOD"                       "HPRD"                       </span></span>
<span><span class="co">##  [5] "HPRD_MIMP"                   "KEA"                        </span></span>
<span><span class="co">##  [7] "Li2012"                      "MIMP"                       </span></span>
<span><span class="co">##  [9] "NCI-PID_ProtMapper"          "phosphoELM"                 </span></span>
<span><span class="co">## [11] "phosphoELM_MIMP"             "PhosphoNetworks"            </span></span>
<span><span class="co">## [13] "PhosphoSite"                 "PhosphoSite_MIMP"           </span></span>
<span><span class="co">## [15] "PhosphoSite_ProtMapper"      "ProtMapper"                 </span></span>
<span><span class="co">## [17] "REACH_ProtMapper"            "Reactome_ProtMapper"        </span></span>
<span><span class="co">## [19] "RLIMS-P_ProtMapper"          "SIGNOR"                     </span></span>
<span><span class="co">## [21] "SIGNOR_ProtMapper"           "Sparser_ProtMapper"</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store the enzyme-PTM interactions into a dataframe.</span></span>
<span><span class="co">## No filtering by databases in this case.</span></span>
<span><span class="va">enzsub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We can select and print the reactions between a specific kinase and</span></span>
<span><span class="co">## a specific substrate</span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">enzsub</span>,</span>
<span>    <span class="va">enzyme_genesymbol</span> <span class="op">==</span> <span class="st">"MAP2K1"</span>,</span>
<span>    <span class="va">substrate_genesymbol</span> <span class="op">==</span> <span class="st">"MAPK3"</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Unknown or uninitialised column: `is_stimulation`.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">##   enzyme          interaction substrate           modification    n_resources</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> MAP2K1 (Q02750) ====&gt;       MAPK3_Y204 (P27361) phosphorylation           8</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> MAP2K1 (Q02750) ====&gt;       MAPK3_T202 (P27361) phosphorylation           8</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> MAP2K1 (Q02750) ====&gt;       MAPK3_T207 (P27361) phosphorylation           2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> MAP2K1 (Q02750) ====&gt;       MAPK3_Y210 (P27361) phosphorylation           2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> MAP2K1 (Q02750) ====&gt;       MAPK3_T80 (P27361)  phosphorylation           1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> MAP2K1 (Q02750) ====&gt;       MAPK3_Y222 (P27361) phosphorylation           1</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## In the previous results, we can see that enzyme-PTM relationships do not</span></span>
<span><span class="co">## contain sign (activation/inhibition). We can generate this information</span></span>
<span><span class="co">## based on the protein-protein OmniPath interaction dataset.</span></span>
<span><span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_interactions.html">import_omnipath_interactions</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enzsub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_signed_ptms.html">get_signed_ptms</a></span><span class="op">(</span><span class="va">enzsub</span>, <span class="va">interactions</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We select again the same kinase and substrate. Now we have information</span></span>
<span><span class="co">## about inhibition or activation when we print the enzyme-PTM relationships</span></span>
<span><span class="fu"><a href="../reference/print_interactions.html">print_interactions</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">enzsub</span>,<span class="va">enzyme_genesymbol</span><span class="op">==</span><span class="st">"MAP2K1"</span>,</span>
<span>    <span class="va">substrate_genesymbol</span><span class="op">==</span><span class="st">"MAPK3"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            enzyme interaction           substrate    modification n_resources</span></span>
<span><span class="co">## 3 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y204 (P27361) phosphorylation           8</span></span>
<span><span class="co">## 1 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_T202 (P27361) phosphorylation           8</span></span>
<span><span class="co">## 4 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y210 (P27361) phosphorylation           2</span></span>
<span><span class="co">## 6 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_T207 (P27361) phosphorylation           2</span></span>
<span><span class="co">## 2 MAP2K1 (Q02750)  ==( + )==&gt;  MAPK3_T80 (P27361) phosphorylation           1</span></span>
<span><span class="co">## 5 MAP2K1 (Q02750)  ==( + )==&gt; MAPK3_Y222 (P27361) phosphorylation           1</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We can also transform the enzyme-PTM relationships into a graph.</span></span>
<span><span class="va">enzsub_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enzsub_graph.html">enzsub_graph</a></span><span class="op">(</span>enzsub <span class="op">=</span> <span class="va">enzsub</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We download PTMs for mouse</span></span>
<span><span class="va">enzsub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_enzsub.html">import_omnipath_enzsub</a></span><span class="op">(</span></span>
<span>    resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PhosphoSite"</span>, <span class="st">"SIGNOR"</span><span class="op">)</span>,</span>
<span>    organism <span class="op">=</span> <span class="fl">10090</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="complexes">Complexes<a class="anchor" aria-label="anchor" href="#complexes"></a>
</h3>
<p>Some studies indicate that around 80% of the human proteins operate in
complexes, and many proteins belong to several different complexes
<span class="citation">(Berggård, Linse, and James 2007)</span>. These complexes play critical roles in a large variety of
biological processes. Some well-known examples are the proteasome and the
ribosome. Thus, the description of the full set of protein complexes
functioning in cells is essential to improve our understanding of biological
processes.</p>
<p>The <strong>complexes</strong> query provides access to more than 20000 protein
complexes. This comprehensive database has been created by integrating
different resources. We now download these molecular complexes filtering by
some of the source databases. We check the complexes where a couple of
specific genes participate. First, we look for the complexes where any of
these two genes participate. We then identify the complex where these two
genes are jointly involved. Finally, we perform an enrichment analysis with
the genes taking part in that complex. You should keep an eye on this complex
since it will be used again in the forthcoming sections.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We check the different complexes databases</span></span>
<span><span class="fu"><a href="../reference/get_complex_resources.html">get_complex_resources</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "CellChatDB"     "Cellinker"      "CellPhoneDB"    "CFinder"       </span></span>
<span><span class="co">##  [5] "Compleat"       "ComplexPortal"  "CORUM"          "Guide2Pharma"  </span></span>
<span><span class="co">##  [9] "Havugimana2012" "hu.MAP"         "hu.MAP2"        "ICELLNET"      </span></span>
<span><span class="co">## [13] "KEGG-MEDICUS"   "NetworkBlast"   "PDB"            "SIGNOR"        </span></span>
<span><span class="co">## [17] "SPIKE"</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We query and store complexes from some sources into a dataframe.</span></span>
<span><span class="va">complexes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_complexes.html">import_omnipath_complexes</a></span><span class="op">(</span>resources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CORUM"</span>, <span class="st">"hu.MAP"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We check all the molecular complexes where a set of genes participate</span></span>
<span><span class="va">query_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WRN"</span>, <span class="st">"PARP1"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Complexes where any of the input genes participate</span></span>
<span><span class="va">wrn_parp1_complexes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/get_complex_genes.html">get_complex_genes</a></span><span class="op">(</span></span>
<span>        <span class="va">complexes</span>,</span>
<span>        <span class="va">query_genes</span>,</span>
<span>        total_match <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We print the components of the different selected components</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">wrn_parp1_complexes</span><span class="op">$</span><span class="va">components_genesymbols</span>, <span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "NCAPD2_NCAPG_NCAPH_PARP1_SMC2_SMC4_XRCC1"                             </span></span>
<span><span class="co">## [2] "CCNA2_CDK2_LIG1_PARP1_POLA1_POLD1_POLE_RFC1_RFC2_RPA1_RPA2_RPA3_TOP1" </span></span>
<span><span class="co">## [3] "CCNA2_CCNB1_CDK1_PARP1_POLA1_POLD1_POLE_RFC1_RFC2_RPA1_RPA2_RPA3_TOP1"</span></span>
<span><span class="co">## [4] "MRE11_PARP1_RAD50_TERF2_TERF2IP_XRCC5_XRCC6"                          </span></span>
<span><span class="co">## [5] "TERF2_WRN"                                                            </span></span>
<span><span class="co">## [6] "CALR_DHX30_H2AX_H2BC26_HSPA5_NPM1_PARP1"</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Complexes where all the input genes participate jointly</span></span>
<span><span class="va">complexes_query_genes_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/get_complex_genes.html">get_complex_genes</a></span><span class="op">(</span></span>
<span>        <span class="va">complexes</span>,<span class="va">query_genes</span>,</span>
<span>        total_match<span class="op">=</span><span class="cn">TRUE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We print the components of the different selected components</span></span>
<span><span class="va">complexes_query_genes_join</span><span class="op">$</span><span class="va">components_genesymbols</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "PARP1_WRN_XRCC5_XRCC6"</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wrn_parp1_cplx_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">wrn_parp1_complexes</span><span class="op">$</span><span class="va">components_genesymbols</span>, <span class="st">"_"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We can perform an enrichment analyses with the genes in the complex</span></span>
<span><span class="va">enrichment</span> <span class="op">&lt;-</span> <span class="fu">gprofiler2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gost</a></span><span class="op">(</span></span>
<span>    <span class="va">wrn_parp1_cplx_genes</span>,</span>
<span>    significant <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    user_threshold <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    correction_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>    sources <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:BP"</span>, <span class="st">"GO:CC"</span>, <span class="st">"GO:MF"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We show the most significant results</span></span>
<span><span class="va">enrichment</span><span class="op">$</span><span class="va">result</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">term_id</span>, <span class="va">source</span>, <span class="va">term_name</span>, <span class="va">p_value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">5</span>, <span class="op">-</span><span class="va">p_value</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      term_id source               term_name      p_value</span></span>
<span><span class="co">## 1 GO:0051276  GO:BP chromosome organization 1.488106e-31</span></span>
<span><span class="co">## 2 GO:0006259  GO:BP   DNA metabolic process 2.684684e-30</span></span>
<span><span class="co">## 3 GO:0006281  GO:BP              DNA repair 1.463364e-26</span></span>
<span><span class="co">## 4 GO:0005694  GO:CC              chromosome 1.078139e-29</span></span>
<span><span class="co">## 5 GO:0005654  GO:CC             nucleoplasm 4.198822e-27</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="annotations">Annotations<a class="anchor" aria-label="anchor" href="#annotations"></a>
</h3>
<p>Biological annotations are statements, usually traceable and curated, about
the different features of a biological entity. At the genetic level,
annotations describe the biological function, the subcellular situation,
the DNA location and many other related properties of a particular gene or
its gene products.</p>
<p>The annotations query provides a large variety of data about proteins
and complexes. These data come from dozens of databases and each kind of
annotation record contains different fields. Because of this, here we have a
record_id field which is unique within the records of each database. Each row
contains one key value pair and you need to use the record_id to connect the
related key-value pairs (see examples below).</p>
<p>Now, we focus in the annotations of the complex studied in the previous
section. We first inspect the different available databases in the omnipath
webserver. Then, we download the annotations for our complex itself as a
biological entity. We find annotations related to the nucleus and
transcriptional control, which is in agreement with the enrichment analysis
results of its individual components.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We check the different annotation databases</span></span>
<span><span class="fu"><a href="../reference/get_annotation_resources.html">get_annotation_resources</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "Adhesome"             "Almen2009"            "Baccin2019"          </span></span>
<span><span class="co">##  [4] "CancerDrugsDB"        "CancerGeneCensus"     "CancerSEA"           </span></span>
<span><span class="co">##  [7] "CellCall"             "CellCellInteractions" "CellChatDB"          </span></span>
<span><span class="co">## [10] "CellChatDB_complex"   "Cellinker"            "Cellinker_complex"   </span></span>
<span><span class="co">## [13] "CellPhoneDB"          "CellPhoneDB_complex"  "CellTalkDB"          </span></span>
<span><span class="co">## [16] "CellTypist"           "ComPPI"               "connectomeDB2020"    </span></span>
<span><span class="co">## [19] "CORUM_Funcat"         "CORUM_GO"             "CSPA"                </span></span>
<span><span class="co">## [22] "CSPA_celltype"        "CytoSig"              "DGIdb"               </span></span>
<span><span class="co">## [25] "DisGeNet"             "EMBRACE"              "Exocarta"            </span></span>
<span><span class="co">## [28] "GO_Intercell"         "GPCRdb"               "Guide2Pharma"        </span></span>
<span><span class="co">## [31] "HGNC"                 "HPA_secretome"        "HPA_subcellular"     </span></span>
<span><span class="co">## [34] "HPA_tissue"           "HPMR"                 "HumanCellMap"        </span></span>
<span><span class="co">## [37] "ICELLNET"             "ICELLNET_complex"     "Integrins"           </span></span>
<span><span class="co">## [40] "InterPro"             "IntOGen"              "iTALK"               </span></span>
<span><span class="co">## [43] "KEGG-PC"              "kinase.com"           "Kirouac2010"         </span></span>
<span><span class="co">## [46] "Lambert2018"          "LOCATE"               "LRdb"                </span></span>
<span><span class="co">## [49] "Matrisome"            "MatrixDB"             "MCAM"                </span></span>
<span><span class="co">## [52] "Membranome"           "MSigDB"               "NetPath"             </span></span>
<span><span class="co">## [55] "OPM"                  "PanglaoDB"            "Phobius"             </span></span>
<span><span class="co">## [58] "Phosphatome"          "PROGENy"              "Ramilowski_location" </span></span>
<span><span class="co">## [61] "Ramilowski2015"       "scConnect"            "scConnect_complex"   </span></span>
<span><span class="co">## [64] "SignaLink_function"   "SignaLink_pathway"    "SIGNOR"              </span></span>
<span><span class="co">## [67] "Surfaceome"           "talklr"               "TCDB"                </span></span>
<span><span class="co">## [70] "TFcensus"             "TopDB"                "UniProt_family"      </span></span>
<span><span class="co">## [73] "UniProt_keyword"      "UniProt_location"     "UniProt_tissue"      </span></span>
<span><span class="co">## [76] "UniProt_topology"     "Vesiclepedia"         "Wang"                </span></span>
<span><span class="co">## [79] "Zhong2015"</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We can further investigate the features of the complex selected</span></span>
<span><span class="co">## in the previous section.</span></span>
<span></span>
<span><span class="co">## We first get the annotations of the complex itself:</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"COMPLEX:"</span>, <span class="va">wrn_parp1_complexes</span><span class="op">$</span><span class="va">components_genesymbols</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="va">source</span>, <span class="va">label</span>, <span class="va">value</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 10 × 3</span></span></span>
<span><span class="co">##    source       label        value      </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> HumanCellMap localization nucleoplasm</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> HumanCellMap method       SAFE       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> HumanCellMap localization chromatin  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> HumanCellMap method       SAFE       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ComPPI       location     nucleus    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ComPPI       location     nucleus    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ComPPI       location     nucleus    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ComPPI       location     nucleus    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ComPPI       location     cytosol    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ComPPI       location     nucleus</span></span></code></pre>
<p>Afterwards, we explore the annotations of the individual components of the
complex in some databases. We check the pathways where these proteins
are involved. Once again, we also find many nucleus related annotations when
checking their cellular location.</p>
<p>Then, we explore some annotations of its individual components. Pathways
where the proteins belong:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="va">wrn_parp1_cplx_genes</span>,</span>
<span>    resources <span class="op">=</span> <span class="st">"NetPath"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="va">genesymbol</span>, <span class="va">value</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 86 × 2</span></span></span>
<span><span class="co">##    genesymbol value                                              </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                              </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> CDK2       Interleukin-2 (IL-2)                               </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> CDK2       Epidermal growth factor receptor (EGFR)            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> CDK2       Thyroid-stimulating hormone (TSH)                  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> CDK2       Kit Receptor                                       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> CDK2       Notch                                              </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> CDK2       B cell receptor (BCR)                              </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> CDK2       Oncostatin-M (OSM)                                 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> CDK2       Advanced glycation end-products (AGE/RAGE)         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> CDK2       Transforming growth factor beta (TGF-beta) receptor</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> CDK2       Inhibitor of differentiation (ID)                  </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 76 more rows</span></span></span></code></pre>
<p>Subcellular localization of our proteins:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="va">wrn_parp1_cplx_genes</span>,</span>
<span>    resources <span class="op">=</span> <span class="st">"ComPPI"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Since we have same record_id for some results of our query, we spread these
records across columns:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/spread.html" class="external-link">spread</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="va">label</span>, <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">10</span>, <span class="va">score</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 10 × 7</span></span></span>
<span><span class="co">##    uniprot genesymbol entity_type source record_id location score             </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> P35580  MYH10      protein     ComPPI     <span style="text-decoration: underline;">10</span>158 cytosol  0.9999999999638732</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Q9NYB0  TERF2IP    protein     ComPPI      <span style="text-decoration: underline;">4</span>117 nucleus  0.9999999999187149</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> P15927  RPA2       protein     ComPPI      <span style="text-decoration: underline;">4</span>343 nucleus  0.999999999827968 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Q15554  TERF2      protein     ComPPI       854 nucleus  0.9999999956992   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> P16104  H2AX       protein     ComPPI     <span style="text-decoration: underline;">17</span>438 nucleus  0.9999999922553344</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> P09884  POLA1      protein     ComPPI     <span style="text-decoration: underline;">20</span>949 nucleus  0.999999990784    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> O95347  SMC2       protein     ComPPI     <span style="text-decoration: underline;">14</span>574 nucleus  0.9999999903232   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> P12956  XRCC6      protein     ComPPI      <span style="text-decoration: underline;">2</span>961 nucleus  0.99999997629184  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> P78527  PRKDC      protein     ComPPI     <span style="text-decoration: underline;">13</span>352 nucleus  0.999999951616    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Q15021  NCAPD2     protein     ComPPI     <span style="text-decoration: underline;">14</span>405 nucleus  0.9999999491968</span></span></code></pre>
<p>The way above, we more or less reconstituted the data as it is in the
original resource. The same can be done much easier by passing the
<code>wide = TRUE</code> parameter to <code>import_omnipath_annotations</code>. In this case,
if the data contains more than one resources, a list of data frames
will be returned.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signaling_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_annotations.html">import_omnipath_annotations</a></span><span class="op">(</span></span>
<span>    resources <span class="op">=</span> <span class="st">"SignaLink_pathway"</span>,</span>
<span>    wide <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="intercell">Intercell<a class="anchor" aria-label="anchor" href="#intercell"></a>
</h3>
<p>Cells perceive cues from their microenvironment and neighboring cells, and
respond accordingly to ensure proper activities and coordination between
them. The ensemble of these communication process is called inter-cellular
signaling (<strong>intercell</strong>).</p>
<p><strong>Intercell</strong> query provides information about the roles of proteins in
inter-cellular signaling (e.g. if a protein is a ligand, a receptor, an
extracellular matrix (ECM) component, etc.) This query type is very similar to
annotations. However, <strong>intercell</strong> data does not come from original
sources, but combined from several databases by us into categories (we also
refer to the original sources).</p>
<p>We first inspect the different categories available in the OmniPath webserver.
Then, we focus again in our previously selected complex and we check its
the location of its individual components in the inter-cellular context. We
can however see that the components of this particular complex are
intracellular.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We check some of the different intercell categories</span></span>
<span><span class="fu"><a href="../reference/get_intercell_generic_categories.html">get_intercell_generic_categories</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "transmembrane"                       "transmembrane_predicted"            </span></span>
<span><span class="co">##  [3] "peripheral"                          "plasma_membrane"                    </span></span>
<span><span class="co">##  [5] "plasma_membrane_transmembrane"       "plasma_membrane_regulator"          </span></span>
<span><span class="co">##  [7] "plasma_membrane_peripheral"          "secreted"                           </span></span>
<span><span class="co">##  [9] "cell_surface"                        "ecm"                                </span></span>
<span><span class="co">## [11] "ligand"                              "receptor"                           </span></span>
<span><span class="co">## [13] "secreted_enzyme"                     "secreted_peptidase"                 </span></span>
<span><span class="co">## [15] "extracellular"                       "intracellular"                      </span></span>
<span><span class="co">## [17] "receptor_regulator"                  "secreted_receptor"                  </span></span>
<span><span class="co">## [19] "sparc_ecm_regulator"                 "ecm_regulator"                      </span></span>
<span><span class="co">## [21] "ligand_regulator"                    "cell_surface_ligand"                </span></span>
<span><span class="co">## [23] "cell_adhesion"                       "matrix_adhesion"                    </span></span>
<span><span class="co">## [25] "adhesion"                            "matrix_adhesion_regulator"          </span></span>
<span><span class="co">## [27] "cell_surface_enzyme"                 "cell_surface_peptidase"             </span></span>
<span><span class="co">## [29] "secreted_enyzme"                     "extracellular_peptidase"            </span></span>
<span><span class="co">## [31] "secreted_peptidase_inhibitor"        "transporter"                        </span></span>
<span><span class="co">## [33] "ion_channel"                         "ion_channel_regulator"              </span></span>
<span><span class="co">## [35] "gap_junction"                        "tight_junction"                     </span></span>
<span><span class="co">## [37] "adherens_junction"                   "desmosome"                          </span></span>
<span><span class="co">## [39] "intracellular_intercellular_related"</span></span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We import the intercell data into a dataframe</span></span>
<span><span class="va">intercell</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_omnipath_intercell.html">import_omnipath_intercell</a></span><span class="op">(</span></span>
<span>    scope <span class="op">=</span> <span class="st">"generic"</span>,</span>
<span>    aspect <span class="op">=</span> <span class="st">"locational"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We check the intercell annotations for the individual components of</span></span>
<span><span class="co">## our previous complex. We filter our data to print it in a good format</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">intercell</span>, <span class="va">genesymbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">wrn_parp1_cplx_genes</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">genesymbol</span>, <span class="va">parent</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">category</span>, <span class="va">genesymbol</span>, <span class="va">parent</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">genesymbol</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 80 × 3</span></span></span>
<span><span class="co">##    category        genesymbol parent         </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> extracellular   ACTB       extracellular  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> intracellular   ACTB       intracellular  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> intracellular   ATE1       intracellular  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> intracellular   BANF1      intracellular  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> secreted        CALR       secreted       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> extracellular   CALR       extracellular  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> intracellular   CALR       intracellular  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> transmembrane   CAMK2D     transmembrane  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> plasma_membrane CAMK2D     plasma_membrane</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> intracellular   CAMK2D     intracellular  </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 70 more rows</span></span></span></code></pre>
<p>The <code>import_intercell_network</code> function creates the most complete network,
including many interactions which are false positives in the context of
interacellular communication. It is highly recommended to apply some
quality filtering on this network. The <code>high_confidence</code> parameter performs
a quiet stringent filtering:</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">icn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_intercell_network.html">import_intercell_network</a></span><span class="op">(</span>high_confidence <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Using the function <code>filter_intercell_network</code> instead, you have much more
flexibility to adjust the stringency of the filtering to the needs of your
analysis. See the full list of options in the docs of the function.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">icn</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="../reference/import_intercell_network.html">import_intercell_network</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filter_intercell_network.html">filter_intercell_network</a></span><span class="op">(</span></span>
<span>        min_curation_effort <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        consensus_percentile <span class="op">=</span> <span class="fl">33</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We close graphical connections</span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.list</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>  <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em> provides access to the wealth of data stored in the
OmniPath webservice <a href="http://omnipathdb.org/" class="external-link uri">http://omnipathdb.org/</a> from the <em>R</em> enviroment.
In addition, it contains some utility functions for visualization, filtering
and analysis. The main strength of <em><a href="https://bioconductor.org/packages/3.18/OmnipathR" class="external-link">OmnipathR</a></em> is the
straightforward transformation of the different OmniPath data into commonly
used <em>R</em> objects, such as dataframes and graphs. Consequently, it allows an
easy integration of the different types of data and a gateway to the vast
number of <em>R</em> packages dedicated to the analysis and representaiton of
biological data. We highlighted these abilities in some of the examples
detailed in previous sections of this document.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Arch Linux</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/libblas.so.3.12.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/liblapack.so.3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Madrid</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] gprofiler2_0.2.3 tidyr_1.3.1      igraph_2.0.3     OmnipathR_3.13.1</span></span>
<span><span class="co">## [5] knitr_1.45      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] gtable_0.3.4        xfun_0.42           bslib_0.6.1        </span></span>
<span><span class="co">##  [4] ggplot2_3.5.0       htmlwidgets_1.6.4   tzdb_0.4.0         </span></span>
<span><span class="co">##  [7] bitops_1.0-7        vctrs_0.6.5         tools_4.3.3        </span></span>
<span><span class="co">## [10] generics_0.1.3      curl_5.2.1          tibble_3.2.1       </span></span>
<span><span class="co">## [13] fansi_1.0.6         highr_0.10          pkgconfig_2.0.3    </span></span>
<span><span class="co">## [16] data.table_1.15.2   checkmate_2.3.1     readxl_1.4.3       </span></span>
<span><span class="co">## [19] lifecycle_1.0.4     compiler_4.3.3      stringr_1.5.1      </span></span>
<span><span class="co">## [22] progress_1.2.3      munsell_0.5.0       BiocStyle_2.30.0   </span></span>
<span><span class="co">## [25] htmltools_0.5.7     sass_0.4.9          RCurl_1.98-1.14    </span></span>
<span><span class="co">## [28] lazyeval_0.2.2      yaml_2.3.8          plotly_4.10.4      </span></span>
<span><span class="co">## [31] later_1.3.2         pillar_1.9.0        crayon_1.5.2       </span></span>
<span><span class="co">## [34] jquerylib_0.1.4     cachem_1.0.8        tidyselect_1.2.1   </span></span>
<span><span class="co">## [37] rvest_1.0.4         zip_2.3.0           digest_0.6.35      </span></span>
<span><span class="co">## [40] stringi_1.8.3       dplyr_1.1.4         purrr_1.0.2        </span></span>
<span><span class="co">## [43] bookdown_0.38       fastmap_1.1.1       grid_4.3.3         </span></span>
<span><span class="co">## [46] colorspace_2.1-0    cli_3.6.2           logger_0.3.0       </span></span>
<span><span class="co">## [49] magrittr_2.0.3      XML_3.99-0.16.1     utf8_1.2.4         </span></span>
<span><span class="co">## [52] readr_2.1.5         withr_3.0.0         prettyunits_1.2.0  </span></span>
<span><span class="co">## [55] scales_1.3.0        backports_1.4.1     rappdirs_0.3.3     </span></span>
<span><span class="co">## [58] bit64_4.0.5         lubridate_1.9.3     timechange_0.3.0   </span></span>
<span><span class="co">## [61] rmarkdown_2.26      httr_1.4.7          bit_4.0.5          </span></span>
<span><span class="co">## [64] cellranger_1.1.0    png_0.1-8           hms_1.1.3          </span></span>
<span><span class="co">## [67] evaluate_0.23       viridisLite_0.4.2   rlang_1.1.3        </span></span>
<span><span class="co">## [70] Rcpp_1.0.12         glue_1.7.0          selectr_0.4-2      </span></span>
<span><span class="co">## [73] BiocManager_1.30.22 xml2_1.3.6          vroom_1.6.5        </span></span>
<span><span class="co">## [76] jsonlite_1.8.8      R6_2.5.1</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-BenShlomo2003" class="csl-entry">
Ben-Shlomo, I., S. Yu Hsu, R. Rauch, H. W. Kowalski, and A. J. W. Hsueh. 2003. <span>“Signaling Receptome: A Genomic and Evolutionary Perspective of Plasma Membrane Receptors Involved in Signal Transduction.”</span> <em>Science Signaling</em> 2003 (187): re9–9. <a href="https://doi.org/10.1126/stke.2003.187.re9" class="external-link">https://doi.org/10.1126/stke.2003.187.re9</a>.
</div>
<div id="ref-Berggrd2007" class="csl-entry">
Berggård, Tord, Sara Linse, and Peter James. 2007. <span>“Methods for the Detection and Analysis of Protein<span></span>protein Interactions.”</span> <em><span>PROTEOMICS</span></em> 7 (16): 2833–42. <a href="https://doi.org/10.1002/pmic.200700131" class="external-link">https://doi.org/10.1002/pmic.200700131</a>.
</div>
<div id="ref-Dinkel2010" class="csl-entry">
Dinkel, H., C. Chica, A. Via, C. M. Gould, L. J. Jensen, T. J. Gibson, and F. Diella. 2010. <span>“Phospho.<span>ELM</span>: A Database of Phosphorylation Sites–Update 2011.”</span> <em>Nucleic Acids Research</em> 39 (Database): D261–67. <a href="https://doi.org/10.1093/nar/gkq1104" class="external-link">https://doi.org/10.1093/nar/gkq1104</a>.
</div>
<div id="ref-Dobson2014" class="csl-entry">
Dobson, László, Tamás Langó, István Reményi, and Gábor E. Tusnády. 2014. <span>“Expediting Topology Data Gathering for the <span>TOPDB</span> Database.”</span> <em>Nucleic Acids Research</em> 43 (D1): D283–89. <a href="https://doi.org/10.1093/nar/gku1119" class="external-link">https://doi.org/10.1093/nar/gku1119</a>.
</div>
<div id="ref-Drew2017" class="csl-entry">
Drew, Kevin, Chanjae Lee, Ryan L Huizar, Fan Tu, Blake Borgeson, Claire D McWhite, Yun Ma, John B Wallingford, and Edward M Marcotte. 2017. <span>“Integration of over 9, 000 Mass Spectrometry Experiments Builds a Global Map of Human Protein Complexes.”</span> <em>Molecular Systems Biology</em> 13 (6): 932. <a href="https://doi.org/10.15252/msb.20167490" class="external-link">https://doi.org/10.15252/msb.20167490</a>.
</div>
<div id="ref-GarciaAlonso2019" class="csl-entry">
Garcia-Alonso, Luz, Christian H. Holland, Mahmoud M. Ibrahim, Denes Turei, and Julio Saez-Rodriguez. 2019. <span>“Benchmark and Integration of Resources for the Estimation of Human Transcription Factor Activities.”</span> <em>Genome Research</em> 29 (8): 1363–75. <a href="https://doi.org/10.1101/gr.240663.118" class="external-link">https://doi.org/10.1101/gr.240663.118</a>.
</div>
<div id="ref-Giurgiu2018" class="csl-entry">
Giurgiu, Madalina, Julian Reinhard, Barbara Brauner, Irmtraud Dunger-Kaltenbach, Gisela Fobo, Goar Frishman, Corinna Montrone, and Andreas Ruepp. 2018. <span>“<span>CORUM</span>: The Comprehensive Resource of Mammalian Protein Complexes<span></span>2019.”</span> <em>Nucleic Acids Research</em> 47 (D1): D559–63. <a href="https://doi.org/10.1093/nar/gky973" class="external-link">https://doi.org/10.1093/nar/gky973</a>.
</div>
<div id="ref-Hornbeck2014" class="csl-entry">
Hornbeck, Peter V., Bin Zhang, Beth Murray, Jon M. Kornhauser, Vaughan Latham, and Elzbieta Skrzypek. 2014. <span>“<span>PhosphoSitePlus</span>, 2014: Mutations, <span>PTMs</span> and Recalibrations.”</span> <em>Nucleic Acids Research</em> 43 (D1): D512–20. <a href="https://doi.org/10.1093/nar/gku1267" class="external-link">https://doi.org/10.1093/nar/gku1267</a>.
</div>
<div id="ref-Keerthikumar2016" class="csl-entry">
Keerthikumar, Shivakumar, David Chisanga, Dinuka Ariyaratne, Haidar Al Saffar, Sushma Anand, Kening Zhao, Monisha Samuel, et al. 2016. <span>“<span>ExoCarta</span>: A Web-Based Compendium of Exosomal Cargo.”</span> <em>Journal of Molecular Biology</em> 428 (4): 688–92. <a href="https://doi.org/10.1016/j.jmb.2015.09.019" class="external-link">https://doi.org/10.1016/j.jmb.2015.09.019</a>.
</div>
<div id="ref-Turei2016" class="csl-entry">
Türei, Dénes, Tamás Korcsmáros, and Julio Saez-Rodriguez. 2016. <span>“<span>OmniPath</span>: Guidelines and Gateway for Literature-Curated Signaling Pathway Resources.”</span> <em>Nature Methods</em> 13 (12): 966–67. <a href="https://doi.org/10.1038/nmeth.4077" class="external-link">https://doi.org/10.1038/nmeth.4077</a>.
</div>
<div id="ref-VentoTormo2018" class="csl-entry">
Vento-Tormo, Roser, Mirjana Efremova, Rachel A. Botting, Margherita Y. Turco, Miquel Vento-Tormo, Kerstin B. Meyer, Jong-Eun Park, et al. 2018. <span>“Single-Cell Reconstruction of the Early Maternal<span></span>fetal Interface in Humans.”</span> <em>Nature</em> 563 (7731): 347–53. <a href="https://doi.org/10.1038/s41586-018-0698-6" class="external-link">https://doi.org/10.1038/s41586-018-0698-6</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alberto Valdeolivas, Denes Turei, Attila Gabor, Diego Mananes, Aurelien Dugourd.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
